<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=windows-1252">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 11">
<meta name=Originator content="Microsoft Word 11">
<link rel=File-List href="blatSpecs_files/filelist.xml">
<script type="text/javascript" src="/js/jquery.js"></script>
<script type="text/javascript" src="/js/staticDoc.js"></script>
<title>BLAT Program Specifications</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Jim Kent</o:Author>
  <o:LastAuthor>Soe</o:LastAuthor>
  <o:Revision>2</o:Revision>
  <o:TotalTime>106</o:TotalTime>
  <o:LastPrinted>2003-06-11T22:46:00Z</o:LastPrinted>
  <o:Created>2007-08-10T00:02:00Z</o:Created>
  <o:LastSaved>2007-08-10T00:02:00Z</o:LastSaved>
  <o:Pages>1</o:Pages>
  <o:Words>4374</o:Words>
  <o:Characters>24938</o:Characters>
  <o:Company> </o:Company>
  <o:Lines>207</o:Lines>
  <o:Paragraphs>58</o:Paragraphs>
  <o:CharactersWithSpaces>29254</o:CharactersWithSpaces>
  <o:Version>11.6568</o:Version>
 </o:DocumentProperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:SpellingState>Clean</w:SpellingState>
  <w:DisplayHorizontalDrawingGridEvery>0</w:DisplayHorizontalDrawingGridEvery>
  <w:DisplayVerticalDrawingGridEvery>0</w:DisplayVerticalDrawingGridEvery>
  <w:UseMarginsForDrawingGridOrigin/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:Compatibility>
   <w:FootnoteLayoutLikeWW8/>
   <w:ShapeLayoutLikeWW8/>
   <w:AlignTablesRowByRow/>
   <w:ForgetLastTabAlignment/>
   <w:LayoutRawTableWidth/>
   <w:LayoutTableRowsApart/>
   <w:UseWord97LineBreakingRules/>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:UseWord2002TableStyleRules/>
  </w:Compatibility>
  <w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel>
 </w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" LatentStyleCount="156">
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Courier;
	panose-1:2 7 4 9 2 2 5 2 4 4;
	mso-font-charset:0;
	mso-generic-font-family:modern;
	mso-font-format:other;
	mso-font-pitch:fixed;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;
	mso-font-charset:2;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:0 268435456 0 0 -2147483648 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:1627421319 -2147483648 8 0 66047 0;}
@font-face
	{font-family:"Fang Song";
	mso-font-charset:80;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:16777216 0 235143169 0 262144 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";}
h2
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:2;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-font-weight:normal;
	font-style:italic;
	mso-bidi-font-style:normal;}
h3
	{mso-style-next:Normal;
	margin-top:12.0pt;
	margin-right:0in;
	margin-bottom:3.0pt;
	margin-left:0in;
	mso-pagination:widow-orphan;
	page-break-after:avoid;
	mso-outline-level:3;
	font-size:12.0pt;
	mso-bidi-font-size:10.0pt;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";
	font-weight:normal;}
p.MsoBodyTextIndent, li.MsoBodyTextIndent, div.MsoBodyTextIndent
	{margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:193.7pt;
	margin-bottom:.0001pt;
	text-indent:-193.7pt;
	mso-pagination:widow-orphan;
	tab-stops:193.5pt;
	font-size:10.0pt;
	font-family:"Courier New";
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;
	text-underline:single;}
p.MsoDocumentMap, li.MsoDocumentMap, div.MsoDocumentMap
	{margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	background:navy;
	font-size:10.0pt;
	font-family:Tahoma;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
span.SpellE
	{mso-style-name:"";
	mso-spl-e:yes;}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in .8in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 @list l0
	{mso-list-id:125394729;
	mso-list-type:hybrid;
	mso-list-template-ids:-770682052 67698705 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l0:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:348071437;
	mso-list-type:hybrid;
	mso-list-template-ids:-458559594 67698705 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l1:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:1009257236;
	mso-list-type:hybrid;
	mso-list-template-ids:1662435710 67698689 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l2:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
@list l3
	{mso-list-id:1239092154;
	mso-list-type:hybrid;
	mso-list-template-ids:50606982 67698689 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l3:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
@list l4
	{mso-list-id:1292908230;
	mso-list-type:hybrid;
	mso-list-template-ids:724436240 67698705 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l4:level1
	{mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5
	{mso-list-id:1919442949;
	mso-list-type:hybrid;
	mso-list-template-ids:-915084300 67698689 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l5:level1
	{mso-level-number-format:bullet;
	mso-level-text:\F0B7;
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	font-family:Symbol;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ansi-language:#0400;
	mso-fareast-language:#0400;
	mso-bidi-language:#0400;}
</style>
<![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=Section1>

<h2><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt;font-weight:normal;
font-style:normal'>BLAT Suite Program Specifications and User Guide<o:p></o:p></span></h2>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>General:<o:p></o:p></span></h3>

<p class=MsoNormal>Blat produces two major classes of alignments: at the DNA
level between two sequences that are of 95% or greater identity, but which may
include large inserts, and at the protein or translated DNA level between
sequences that are of 80% or greater identity and may also include large
inserts.<span style='mso-spacerun:yes'>  </span>The output of BLAT is
flexible.<span style='mso-spacerun:yes'>  </span>By default it is a simple
tab-delimited file which describes the alignment, but which does not include
the sequence of the alignment itself.<span style='mso-spacerun:yes'> 
</span>Optionally it can produce BLAST and WU-BLAST <span class=SpellE>compatable</span>
output as well as a number of other formats.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The main programs in the blat suite are:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'><span
     class=SpellE>gfServer</span> – a server that maintains an index of the
     genome in memory and uses the index to quickly find regions with high
     levels of sequence similarity to a query sequence.</li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'><span
     class=SpellE>gfClient</span> – a program that queries <span class=SpellE>gfServer</span>
     over the network, and then does a detailed alignment of the query sequence
     with regions found by <span class=SpellE>gfServer</span>.</li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'>blat
     –combines client and server into a single program, first building the
     index, then using the index, and then exiting.<span
     style='mso-spacerun:yes'>  </span></li>
 <li class=MsoNormal style='mso-list:l5 level1 lfo4;tab-stops:list .5in'><span
     class=SpellE>webBlat</span> – a web based version of <span class=SpellE>gfClient</span>
     that presents the alignments in an interactive fashion.</li>
</ul>

<p class=MsoNormal>Building an index of the genome typically takes 10 or 15
minutes. <span style='mso-spacerun:yes'> </span>Typically for interactive
applications one uses <span class=SpellE>gfServer</span> to build a whole
genome index.<span style='mso-spacerun:yes'>  </span>At that point <span
class=SpellE>gfClient</span> or <span class=SpellE>webBlat</span> can align a
single query within few seconds.<span style='mso-spacerun:yes'>  </span>If one
is aligning a lot of sequences in a batch mode then blat can be more efficient,
particularly if run on a cluster of computers.<span style='mso-spacerun:yes'> 
</span>Each blat run is typically done against a single chromosome, but with a
large number of query sequences.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Other programs in the blat suite are:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>pslSort</span> – combines and sorts the output of multiple
     blat runs. <span style='mso-spacerun:yes'> </span>(The blat default output
     format is .<span class=SpellE>psl</span>). </li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>pslReps</span> – selects the best alignments for a particular
     query sequence, using a ‘near best in genome’ approach.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>pslPretty</span> – converts alignments from the <span
     class=SpellE>psl</span> format, which is tab-delimited format and does not
     include the bases themselves, to a more readable alignment format. </li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>faToTwoBit</span> – convert Fasta format sequence files to a
     dense randomly-<span class=SpellE>accessable</span><span
     style='mso-spacerun:yes'>  </span>.2bit format that <span class=SpellE>gfClient</span>
     can use.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>twoBitToFa</span> – convert from the .2bit format back to
     fasta</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>faToNib</span> – convert from Fasta to a somewhat less dense
     randomly accessible format that predates .2bit.<span
     style='mso-spacerun:yes'>  </span>Note each .nib file can only contain a
     single sequence.</li>
 <li class=MsoNormal style='mso-list:l2 level1 lfo5;tab-stops:list .5in'><span
     class=SpellE>nibFrag</span> – convert portions of a nib file back to
     fasta.</li>
</ul>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>In addition you may be interested in the following programs
which are not part of the BLAT suite:</p>

<ul style='margin-top:0in' type=disc>
 <li class=MsoNormal style='mso-list:l3 level1 lfo6;tab-stops:list .5in'>In
     Silico PCR –<span style='mso-spacerun:yes'>  </span>given two primers
     quickly find the sequence between them.<span style='mso-spacerun:yes'> 
     </span>Available from Kent Informatics. This includes <span class=SpellE>webPCR</span>,
     an interface similar to <span class=SpellE>webBlat</span>.</li>
 <li class=MsoNormal style='mso-list:l3 level1 lfo6;tab-stops:list .5in'>The
     Genome Browser – display annotations as a series of tracks on top of the
     genome.<span style='mso-spacerun:yes'>  </span>Available from the
     University of California Santa Cruz.<span style='mso-spacerun:yes'> 
     </span>See <a class="change" href="http://genome.ucsc.edu/license/">http://genome.ucsc.edu/license/</a>.</li>
</ul>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Running<span
style='mso-spacerun:yes'>  </span>the<span style='mso-spacerun:yes'> 
</span>Programs:<o:p></o:p></span></h3>

<p class=MsoNormal>The command line options of each of the programs is described
below. Similar summaries of usage are printed when a command is run with no
arguments.<span style='mso-spacerun:yes'>  </span>See the next section for info
on installing <span class=SpellE>webBlat</span>.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>blat<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>blat
- Standalone BLAT sequence search command line tool<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>blat database query [-<span class=SpellE>ooc</span>=11.ooc]
<span class=SpellE>output.psl</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>database and query are each either a .<span
class=SpellE>fa</span> , .nib or .2bit file,<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>or a list these files one file name per
line.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>ooc</span>=11.ooc tells
the program to load over-occurring 11-mers from<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>and external file.<span
style='mso-spacerun:yes'>  </span>This will increase the speed<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>by a factor of 40 in many cases,
but is not required<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>output.psl</span> is
where to put the output.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>Subranges</span> of nib
and .2bit files may specified using the syntax:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>      </span>/path/<span class=SpellE>file.nib:seqid:start</span>-end<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>or<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>      </span>/path/file.2bit:seqid:start-end<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>or<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>      </span>/path/<span class=SpellE>file.nib:start</span>-end<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>With the second form, a sequence id of <span
class=SpellE>file:start</span>-end will be used.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-t=type<span style='mso-spacerun:yes'>    
</span>Database type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dna</span>
- DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>prot</span>
- protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span><span
style='mso-spacerun:yes'>               </span><span class=SpellE>dnax</span> -
DNA sequence translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is <span
class=SpellE>dna</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-q=type<span style='mso-spacerun:yes'>    
</span>Query type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dna</span>
- DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>rna</span>
- RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>prot</span>
- protein sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dnax</span>
- DNA sequence translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>rnax</span>
- DNA sequence translated in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is <span
class=SpellE>dna</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>prot</span><span
style='mso-spacerun:yes'>       </span>Synonymous with -t=<span class=SpellE>prot</span>
-q=<span class=SpellE>prot</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>ooc</span>=<span
class=SpellE>N.ooc</span><span style='mso-spacerun:yes'>  </span>Use overused
tile file <span class=SpellE>N.ooc</span>.<span style='mso-spacerun:yes'> 
</span>N should correspond to <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>the <span class=SpellE>tileSize</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>tileSize</span>=N sets
the size of match that triggers an alignment.<span style='mso-spacerun:yes'> 
</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Usually between 8 and 12<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 11 for DNA and 5 for
protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>stepSize</span>=N spacing
between tiles. Default is <span class=SpellE>tileSize</span>.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>oneOff</span>=N<span
style='mso-spacerun:yes'>   </span>If set to 1 this allows one mismatch in tile
and still<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>triggers an alignments.<span
style='mso-spacerun:yes'>  </span>Default is 0.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minMatch</span>=N sets
the number of tile matches.<span style='mso-spacerun:yes'>  </span>Usually set
from 2 to 4<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotide, 1
for protein.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minScore</span>=N sets
minimum score.<span style='mso-spacerun:yes'>  </span>This is the matches minus
the <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>mismatches minus some sort of
gap penalty.<span style='mso-spacerun:yes'>  </span>Default is 30<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minIdentity</span>=N
Sets minimum sequence identity (in percent).<span style='mso-spacerun:yes'> 
</span>Default is<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>90 for nucleotide searches, 25
for protein or translated<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>protein searches.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxGap</span>=N<span
style='mso-spacerun:yes'>   </span>sets the size of maximum gap between tiles
in a clump.<span style='mso-spacerun:yes'>  </span>Usually<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>set from 0 to 3.<span
style='mso-spacerun:yes'>  </span>Default is 2. Only <span class=SpellE>relevent</span>
for <span class=SpellE>minMatch</span> &gt; 1.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>noHead</span><span
style='mso-spacerun:yes'>     </span>suppress .<span class=SpellE>psl</span>
header (so it's just a tab-separated file)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>makeOoc</span>=<span
class=SpellE>N.ooc</span> Make overused tile file. Target needs to be complete
genome.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>repMatch</span>=N sets
the number of repetitions of a tile allowed before<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>it is marked as overused.<span
style='mso-spacerun:yes'>  </span>Typically this is 256 for <span class=SpellE>tileSize</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>12, 1024 for tile size 11, 4096
for tile size 10.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 1024.<span
style='mso-spacerun:yes'>  </span>Typically only comes into play with <span
class=SpellE>makeOoc</span>.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Also affected by <span
class=SpellE>stepSize</span>. When <span class=SpellE>stepSize</span> is halved
<span class=SpellE>repMatch</span> is<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>doubled to compensate.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-mask=type<span style='mso-spacerun:yes'> 
</span>Mask out repeats.<span style='mso-spacerun:yes'>  </span>Alignments
won't be started in masked region<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>but may extend through it in
nucleotide searches.<span style='mso-spacerun:yes'>  </span>Masked areas<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>are ignored entirely in protein
or translated searches. Types are<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>lower - mask out lower cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>upper - mask out upper cased
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span>out<span
style='mso-spacerun:yes'>   </span>- mask according to <span class=SpellE>database.out</span>
<span class=SpellE>RepeatMasker</span> .out file<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>file.out</span>
- mask database according to <span class=SpellE>RepeatMasker</span> <span
class=SpellE>file.out</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>qMask</span>=type Mask
out repeats in query sequence.<span style='mso-spacerun:yes'>  </span>Similar
to -mask above but<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>for query rather than target
sequence.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-repeats=type Type is same as mask types
above.<span style='mso-spacerun:yes'>  </span>Repeat bases will not be<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>masked in any way, but matches
in repeat areas will be reported<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>separately from matches in other
areas in the <span class=SpellE>psl</span> output.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minRepDivergence</span>=NN
- minimum percent divergence of repeats to allow <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>them to be unmasked.<span
style='mso-spacerun:yes'>  </span>Default is 15.<span
style='mso-spacerun:yes'>  </span>Only relevant for <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>masking using <span
class=SpellE>RepeatMasker</span> .out files.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dots=N<span style='mso-spacerun:yes'>    
</span>Output dot every N sequences to show program's progress<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>trimT</span><span
style='mso-spacerun:yes'>      </span>Trim leading poly-T<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>noTrimA</span><span
style='mso-spacerun:yes'>    </span>Don't trim trailing poly-A<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>trimHardA</span><span
style='mso-spacerun:yes'>  </span>Remove poly-A tail from <span class=SpellE>qSize</span>
as well as alignments in <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span><span class=SpellE>psl</span>
output<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>fastMap</span><span
style='mso-spacerun:yes'>    </span>Run for fast DNA/DNA remapping - not
allowing <span class=SpellE>introns</span>, <o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>requiring high %ID<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-out=type<span style='mso-spacerun:yes'>  
</span>Controls output file format.<span style='mso-spacerun:yes'>  </span>Type
is one of:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>psl</span>
- Default.<span style='mso-spacerun:yes'>  </span>Tab separated format, no
sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>pslx</span>
- Tab separated format with sequence<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>axt</span>
- <span class=SpellE>blastz</span>-associated <span class=SpellE>axt</span>
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>maf</span>
- <span class=SpellE>multiz</span>-associated <span class=SpellE>maf</span>
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>sim4 - similar to sim4
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>wublast</span>
- similar to <span class=SpellE>wublast</span> format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast8- NCBI blast tabular
format<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast9 - NCBI blast tabular
format with comments<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-fine<span style='mso-spacerun:yes'>      
</span>For high quality mRNAs look harder for small initial and<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>terminal <span class=SpellE>exons</span>.<span
style='mso-spacerun:yes'>  </span>Not recommended for <span class=SpellE>ESTs</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxIntron</span>=N<span
style='mso-spacerun:yes'>  </span>Sets maximum <span class=SpellE>intron</span>
size. Default is 750000<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>extendThroughN</span> -
Allows extension of alignment through large blocks of N's<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>Here are some blat settings for
common usage scenarios:</p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>1) Mapping <span class=SpellE>ESTs</span>
to the genome within the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>ooc</span>=11.ooc<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>2) Mapping full length mRNAs to
the genome in the same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>ooc</span>=11.ooc -fine
-q=<span class=SpellE>rna</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>3) Mapping <span class=SpellE>ESTs</span>
to the genome across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=<span class=SpellE>dnax</span> -t=<span
class=SpellE>dnax</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>4) Mapping mRNA to the genome
across species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=<span class=SpellE>rnax</span> -t=<span
class=SpellE>dnax</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>5) Mapping proteins to the
genome</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=<span class=SpellE>prot</span> -t=<span
class=SpellE>dnax</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>6) Mapping DNA to DNA in the
same species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>ooc</span>=11.ooc -<span
class=SpellE>fastMap</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>7) Mapping DNA from one species
to another species</p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-q=<span class=SpellE>dnax</span> -t=<span
class=SpellE>dnax</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>When mapping DNA from one species to
another the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>query side of the alignment should be cut
up into chunks<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>of 25kb or less for best performance.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<b style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;
mso-bidi-font-size:10.0pt;font-family:"Times New Roman";mso-fareast-font-family:
"Times New Roman";mso-ansi-language:EN-US;mso-fareast-language:EN-US;
mso-bidi-language:AR-SA'><br clear=all style='page-break-before:always'>
</span></b>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'>gfServer</span></b></span><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfServer</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Make a server to quickly find where
DNA occurs in genome.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To set up a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> start
host port file(s)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>Where the files are in .nib or .2bit format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To remove a
server:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> stop host
port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with DNA sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> query
host port <span class=SpellE>probe.fa</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with protein sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> <span
class=SpellE>protQuery</span> host port <span class=SpellE>probe.fa</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To query a
server with translated <span class=SpellE>dna</span> sequence:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> <span
class=SpellE>transQuery</span> host port <span class=SpellE>probe.fa</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To process
one probe <span class=SpellE>fa</span> file against a .nib format genome (not
starting server):<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> direct <span
class=SpellE>probe.fa</span> file(s).nib<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To figure out
usage level<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> status
host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>To get input
file list<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span> files
host port<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>Options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>tileSize</span>=N size
of n-<span class=SpellE>mers</span> to index.<span style='mso-spacerun:yes'> 
</span>Default is 11 for nucleotides, 4 for<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>proteins (or translated
nucleotides).<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>stepSize</span>=N
spacing between tiles. Default is <span class=SpellE>tileSize</span>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minMatch</span>=N Number
of n-<span class=SpellE>mer</span> matches that trigger detailed alignment<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotides, 3
for <span class=SpellE>protiens</span>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxGap</span>=N<span
style='mso-spacerun:yes'>   </span>Number of insertions or deletions allowed
between n-<span class=SpellE>mers</span>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 2 for nucleotides, 0
for <span class=SpellE>protiens</span>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-trans<span style='mso-spacerun:yes'> 
</span>Translate database to protein in 6 frames.<span
style='mso-spacerun:yes'>  </span>Note: it is best<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>           </span>to run this on <span class=SpellE>RepeatMasked</span>
data in this case.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-log=<span class=SpellE>logFile</span> keep
a log file that records server requests.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>seqLog</span><span
style='mso-spacerun:yes'>    </span>Include sequences in log file (not logged
with -<span class=SpellE>syslog</span>)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>syslog</span><span
style='mso-spacerun:yes'>    </span>Log to <span class=SpellE>syslog</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>logFacility</span>=facility
log to the specified <span class=SpellE>syslog</span> facility - default
local0.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-mask<span style='mso-spacerun:yes'>     
</span>Use masking from nib file.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>repMatch</span>=N Number
of occurrences of a tile (<span class=SpellE>nmer</span>) that trigger <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>repeat masking the tile. Default
is 1024.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxDnaHits</span>=N
Maximum number of hits for a <span class=SpellE>dna</span> query that are sent
from the<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>server. Default is 100.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxTransHits</span>=N Maximum
number of hits for a translated query that are <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>sent from the server. Default is
200.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxNtSize</span>=N
Maximum size of <span class=SpellE>untranslated</span> DNA query sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 40000<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxAsSize</span>=N
Maximum size of protein or translated DNA queries<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>Default is 8000<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>canStop</span> If set
then a quit message will actually take down the<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>            </span>server<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>gfClient</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>gfClient</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - A client for the genomic finding
program <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>gfClient</span> host port
<span class=SpellE>nibDir</span> <span class=SpellE>in.fa</span> <span
class=SpellE>out.psl</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>where<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>host is the name of the machine running the <span
class=SpellE>gfServer</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>port is the same as you started the <span
class=SpellE>gfServer</span> with<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>nibDir</span> is the path
of the nib files relative to the current dir<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>       </span>(note these are needed by the client as
well as the server)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>in.fa</span> a <span
class=SpellE>fasta</span> format file.<span style='mso-spacerun:yes'> 
</span>May contain multiple records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>out.psl</span> where to
put the output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-t=type<span style='mso-spacerun:yes'>    
</span>Database type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dna</span>
- DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>prot</span>
- protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dnax</span>
- DNA sequence translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>The default is <span
class=SpellE>dna</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-q=type<span style='mso-spacerun:yes'>    
</span>Query type.<span style='mso-spacerun:yes'>  </span>Type is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dna</span>
- DNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>rna</span>
- RNA sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>prot</span>
- protein sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>dnax</span>
- DNA sequence translated in six frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                 </span><span class=SpellE>rnax</span>
- DNA sequence translated in three frames to protein<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dots=N<span style='mso-spacerun:yes'>  
</span>Output a dot every N query sequences<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>nohead</span><span
style='mso-spacerun:yes'>   </span>Suppresses <span class=SpellE>psl</span>
five line header<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minScore</span>=N sets
minimum score.<span style='mso-spacerun:yes'>  </span>This is twice the matches
minus the <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>mismatches minus some sort of
gap penalty.<span style='mso-spacerun:yes'>  </span>Default is 30<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>minIdentity</span>=N
Sets minimum sequence identity (in percent).<span style='mso-spacerun:yes'> 
</span>Default is<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>90 for nucleotide searches, 25
for protein or translated<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>protein searches.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-out=type<span style='mso-spacerun:yes'>  
</span>Controls output file format.<span style='mso-spacerun:yes'>  </span>Type
is one of:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>psl</span>
- Default.<span style='mso-spacerun:yes'>  </span>Tab separated format without
actual sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>pslx</span>
- Tab separated format with sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>axt</span>
- blastz-associated <span class=SpellE>axt</span> format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>maf</span>
- <span class=SpellE>multiz</span>-associated <span class=SpellE>maf</span>
format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span><span class=SpellE>wublast</span>
- similar to <span class=SpellE>wublast</span> format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                   </span>blast - similar to NCBI
blast format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>maxIntron</span>=N<span
style='mso-spacerun:yes'>  </span>Sets maximum intron size. Default is 750000<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>webBlat</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>webBlat</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> generally is not run from the command
line.<span style='mso-spacerun:yes'>  </span>See ‘Setting Up <span
class=SpellE>webBlat</span> instructions below` for information on this
program.<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>faToTwoBit</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>faToTwoBit</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Convert DNA from <span class=SpellE>fasta</span>
to 2bit format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>faToTwoBit</span> <span
class=SpellE>in.fa</span> [in2.fa in3.fa ...] out.2bit<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>noMask</span> - Ignore
lower-case masking in <span class=SpellE>fa</span> file.<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>twoBitToFa</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>twoBitToFa</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Convert all or part of .2bit file to
<span class=SpellE>fasta</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>twoBitToFa</span>
input.2bit <span class=SpellE>output.fa</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>seq</span>=name -
restrict this to just one sequence<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-start=X<span style='mso-spacerun:yes'> 
</span>- start at given position in sequence (zero-based)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-end=X - end at given position in sequence
(non-inclusive)</span><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'>faToNib</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>faToNib</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Convert from .<span class=SpellE>fa</span>
to .nib format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>faToNib</span> <span
class=SpellE>in.fa</span> <span class=SpellE>out.nib</span><o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>nibFrag</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>nibFrag</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Extract part of a nib file as .<span
class=SpellE>fa</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>nibFrag</span> <span
class=SpellE>file.nib</span> start end strand <span class=SpellE>out.fa</span><o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>pslPretty</span></b></span><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'><o:p></o:p></span></b></p>

<p class=MsoNormal><span class=SpellE><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>pslPretty</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-bidi-font-family:"Times New Roman"'> - Convert PSL to human readable output<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span class=SpellE>pslPretty</span> <span
class=SpellE>in.psl</span> <span class=SpellE>target.lst</span> <span
class=SpellE>query.lst</span> <span class=SpellE>pretty.out</span><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-<span class=SpellE>axt</span> - save in
Scott Schwartz's <span class=SpellE>axt</span> format<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-dot=N Put out a dot every N records<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-long - Don't abbreviate long inserts<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'>It's a really good idea if the <span
class=SpellE>psl</span> file is sorted by target if it contains multiple
targets.<span style='mso-spacerun:yes'>  </span>Otherwise this will be very <span
class=SpellE>very</span> slow.<span style='mso-spacerun:yes'>   </span>The
target and query lists can either be <span class=SpellE>fasta</span> files, nib
files, or a list of <span class=SpellE>fasta</span> and/or nib files one per
line.<span style='mso-spacerun:yes'>  </span>Currently this only handles
nucleotide based <span class=SpellE>psl</span> files.</p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'>pslSort</span></b></span><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'>pslSort</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'> -
merge and sort <span class=SpellE>psCluster</span> .<span class=SpellE>psl</span>
output files<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span><span class=SpellE>pslSort</span> <span
class=SpellE>dirs</span>[1|2] <span class=SpellE>outFile</span> <span
class=SpellE>tempDir</span> <span class=SpellE>inDir</span>(s)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>This will sort all of the .<span
class=SpellE>psl</span> files in the directories<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'>inDirs</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'> in
two stages - first into temporary files in <span class=SpellE>tempDir</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>and second into <span
class=SpellE>outFile</span>.<span style='mso-spacerun:yes'>  </span>The device
on <span class=SpellE>tempDir</span> needs to have<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>enough space (typically
15-20 gigabytes if processing whole genome)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span><span class=SpellE>pslSort</span> g2g[1|2] <span
class=SpellE>outFile</span> <span class=SpellE>tempDir</span> <span
class=SpellE>inDir</span>(s)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>This will sort a genome to
genome alignment, reflecting the<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>alignments across the
diagonal.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Adding 1 or 2 after the <span
class=SpellE>dirs</span> or g2g will limit the program to<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>only the first or second
pass <span class=SpellE>repectively</span> of the sort<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span>-verbose=N Set verbosity level, higher for
more output. Default 1<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>Note for huge files <span
class=SpellE>pslSort</span> will run out of memory.<span
style='mso-spacerun:yes'>  </span>The <span class=SpellE>unix</span> sort
command<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>  </span>sort –k 10 *.<span class=SpellE>psl</span>
&gt; <span class=SpellE>sorted.psl</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>may be preferable in these
situations, though the <span class=SpellE>psl</span> header lines should<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>be removed or avoided with
the –<span class=SpellE>noHead</span> option to blat.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:
"Courier New";mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><b
style='mso-bidi-font-weight:normal'><span style='font-size:12.0pt;mso-bidi-font-size:
10.0pt'>pslReps</span></b></span><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt;font-family:"Courier New";mso-fareast-font-family:"Fang Song";
mso-bidi-font-family:"Times New Roman"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span class=SpellE><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'>pslReps</span></span><span
style='font-size:9.0pt;mso-bidi-font-size:10.0pt;font-family:"Courier New";
mso-fareast-font-family:"Fang Song";mso-bidi-font-family:"Times New Roman"'> - <span
class=SpellE>analyse</span> repeats and generate genome wide best<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>alignments from a sorted
set of local alignments<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>usage:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span><span class=SpellE>pslReps</span> <span
class=SpellE>in.psl</span> <span class=SpellE>out.psl</span> <span
class=SpellE>out.psr</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>where <span class=SpellE>in.psl</span>
is an alignment file generated by <span class=SpellE>psLayout</span> and<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>sorted by <span
class=SpellE>pslSort</span>, <span class=SpellE>out.psl</span> is the best
alignment output<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>and <span class=SpellE>out.psr</span>
contains repeat info<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'>options:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>nohead</span> don't add
PSL header<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>ignoreSize</span> Will
not weigh in favor of larger alignments so much<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>noIntrons</span> Will
not penalize for not having <span class=SpellE>introns</span> when calculating<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>              </span>size factor<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>singleHit</span><span
style='mso-spacerun:yes'>  </span>Takes single best hit, not splitting into
parts<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>minCover</span>=0.N
minimum coverage to output.<span style='mso-spacerun:yes'>  </span>Default is
0.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>ignoreNs</span> Ignore
'N's when calculating <span class=SpellE>minCover</span>.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>minAli</span>=0.N
minimum alignment ratio<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>default is 0.93<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>nearTop</span>=0.N how
much can deviate from top and be taken<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>default is 0.01<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>minNearTopSize</span>=N<span
style='mso-spacerun:yes'>  </span>Minimum size of alignment that is near top<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>               </span>for <span class=SpellE>aligmnent</span>
to be kept.<span style='mso-spacerun:yes'>  </span>Default 30.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>    </span>-<span class=SpellE>coverQSizes</span>=file
Tab-separate file with effective query sizes.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>   </span><span
style='mso-spacerun:yes'>                  </span>When used with -<span
class=SpellE>minCover</span>, this allows <span class=SpellE>polyAs</span><o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-fareast-font-family:
"Fang Song";mso-bidi-font-family:"Times New Roman"'><span
style='mso-spacerun:yes'>                     </span>to be excluded from the
coverage calculation<o:p></o:p></span></p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Setting Up <span
class=SpellE>webBlat</span><o:p></o:p></span></h3>

<p class=MsoNormal>INSTALLING WEBBLAT</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Installing A Web-Based Blat Server involves four major
steps:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>1) Creating sequence
databases.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>2) Running the <span
class=SpellE>gfServer</span> program to create in-memory indexes of the
databases.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>3) Editing the <span
class=SpellE>webBlat.cfg</span> file to tell it what machine and port the <span
class=SpellE>gfServer</span>(s)</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>are running on,
and optionally customizing the <span class=SpellE>webBlat</span> appearance to
users.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'> </span>4) Copying the <span
class=SpellE>webBlat</span> executable and <span class=SpellE>webBlat.cfg</span>
to a directory where the web server can</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span>execute <span
class=SpellE>webBlat</span> as a CGI.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>CREATING SEQUENCE DATABASES</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>You create databases with the program <span class=SpellE>faToTwoBit</span>.
Typically you'll create</p>

<p class=MsoNormal>a separate database for each genome you are indexing.<span
style='mso-spacerun:yes'>  </span>Each database can</p>

<p class=MsoNormal>contain up to four billion bases of sequence in an unlimited
number of</p>

<p class=MsoNormal>records.<span style='mso-spacerun:yes'>  </span>The
databases for <span class=SpellE>webPcr</span> and <span class=SpellE>webBlat</span>
are identical.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The input to <span class=SpellE>faToTwoBit</span> is one or
more <span class=SpellE>fasta</span> format files each of which</p>

<p class=MsoNormal>can contain multiple records.<span
style='mso-spacerun:yes'>  </span>If the sequence contains repeat sequences,</p>

<p class=MsoNormal>as is the case with vertebrates and many plants, the repeat
sequences can</p>

<p class=MsoNormal>be represented in lower case and the other sequence in upper
case.<span style='mso-spacerun:yes'>  </span>The <span class=SpellE>gfServer</span></p>

<p class=MsoNormal>program can be configured to ignore the repeat
sequences.<span style='mso-spacerun:yes'>  </span>The output of</p>

<p class=MsoNormal><span class=SpellE>faToTwoBit</span> is a file which is
designed for fast random access and efficient</p>

<p class=MsoNormal>storage.<span style='mso-spacerun:yes'>  </span>The output
files store four bases per byte.<span style='mso-spacerun:yes'>  </span>They
use a small amount</p>

<p class=MsoNormal>of additional space to store the case of the DNA and to keep
track of runs of</p>

<p class=MsoNormal>N's in the input.<span style='mso-spacerun:yes'> 
</span>Non-N ambiguity codes such as Y and U in the input sequence</p>

<p class=MsoNormal>will be converted to N.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>Here's how a typical installation might create a mouse and a
human genome database:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>cd</span> /data/genomes</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>mkdir</span> <span class=SpellE>twoBit</span></p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>faToTwoBit</span> human/hg16/*.<span class=SpellE>fa</span>
twoBit/hg16.2bit</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>faToTwoBit</span> mouse/mm4/*.<span class=SpellE>fa</span>
twoBit/mm4.2bit</p>

<p class=MsoNormal>There's no need to put all of the databases in the same
directory, but it can</p>

<p class=MsoNormal>simplify bookkeeping.<span style='mso-spacerun:yes'> 
</span></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The databases can also be in the .nib format which was used
with blat and</p>

<p class=MsoNormal><span class=SpellE>gfClient/gfServer</span> until
recently.<span style='mso-spacerun:yes'>  </span>The .nib format only packed 2
bases per</p>

<p class=MsoNormal>byte, and could only handle one record per nib file.<span
style='mso-spacerun:yes'>  </span>Recent versions of blat</p>

<p class=MsoNormal>and related programs can use .2bit files as well.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>CREATING IN-MEMORY INDICES WITH GFSERVER</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The <span class=SpellE>gfServer</span> program creates an
in-memory index of a nucleotide sequence database.<span
style='mso-spacerun:yes'>  </span></p>

<p class=MsoNormal>The index can either be for translated or <span
class=SpellE>untranslated</span> searches.<span style='mso-spacerun:yes'> 
</span>Translated indexes</p>

<p class=MsoNormal>enable protein-based blat queries and use approximately two
bytes per unmasked base </p>

<p class=MsoNormal>in the database.<span style='mso-spacerun:yes'>  </span><span
class=SpellE>Untranslated</span> indexes are used nucleotide-based blat queries
as well</p>

<p class=MsoNormal>as for In-<span class=SpellE>silico</span> PCR.<span
style='mso-spacerun:yes'>  </span>An index for normal blat uses approximately
1/4 byte per base.<span style='mso-spacerun:yes'>  </span>For</p>

<p class=MsoNormal>blat on smaller (primer-sized) queries or for In-<span
class=SpellE>silico</span> PCR a more thorough index that</p>

<p class=MsoNormal>requires 1/2 byte per base is recommended.<span
style='mso-spacerun:yes'>  </span>The <span class=SpellE>gfServer</span> is
memory intensive but typically</p>

<p class=MsoNormal><span class=SpellE>doesn</span> not require a lot of CPU
power.<span style='mso-spacerun:yes'>  </span>Memory permitting multiple <span
class=SpellE>gfServers</span> can be</p>

<p class=MsoNormal>run on the same machine.<span style='mso-spacerun:yes'> 
</span></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>A typical installation might go:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>ssh</span> <span class=SpellE>bigRamMachine</span></p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>cd</span> /data/genomes/<span class=SpellE>twoBit</span></p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>gfServer</span> start <span class=SpellE>bigRamMachine</span>
17779 hg16.2bit &amp;</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>gfServer</span> -trans -mask start <span class=SpellE>bigRamMachine</span>
17778 hg16.2bit &amp;</p>

<p class=MsoNormal>the -trans flag makes a translated index.<span
style='mso-spacerun:yes'>   </span>It will take approximately</p>

<p class=MsoNormal>15 minutes to build an <span class=SpellE>untranslated</span>
index, and 45 minutes to build a</p>

<p class=MsoNormal>translate index.<span style='mso-spacerun:yes'>  </span>To
build an <span class=SpellE>untranslated</span> index to be shared with </p>

<p class=MsoNormal>In-<span class=SpellE>silico</span> PCR do</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>    </span><span
class=SpellE>gfServer</span> -<span class=SpellE>stepSize</span>=5 <span
class=SpellE>bigRamMachine</span> 17779 hg16.2bit &amp;</p>

<p class=MsoNormal>This index will be slightly more sensitive, noticeably so
for small query sequences,</p>

<p class=MsoNormal>with blat.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>EDITING THE WEBBLAT.CFG FILE</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The <span class=SpellE>webBlat.cfg</span> file tells the <span
class=SpellE>webBlat</span> program where to look for <span class=SpellE>gfServers</span>
and</p>

<p class=MsoNormal>for sequence.<span style='mso-spacerun:yes'>  </span>The
basic format of the .<span class=SpellE>cfg</span> file is line oriented with
the</p>

<p class=MsoNormal>first word of the line being a command.<span
style='mso-spacerun:yes'>  </span>Blank lines and lines starting with #</p>

<p class=MsoNormal>are ignored.<span style='mso-spacerun:yes'>  </span>The <span
class=SpellE>webBlat.cfg</span> and <span class=SpellE>webPcr.cfg</span> files
are similar. The <span class=SpellE>webBlat.cfg</span></p>

<p class=MsoNormal>commands are:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span><span class=SpellE>gfServer</span><span
style='mso-spacerun:yes'>  </span>-<span style='mso-spacerun:yes'> 
</span>defines host and port a (<span class=SpellE>untranslated</span>) <span
class=SpellE>gfServer</span> is running on, the </p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span><span
style='mso-tab-count:2'>                             </span>associated sequence
directory, and the name of the database to display in</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>                </span>the <span
class=SpellE>webPcr</span> web page.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span><span class=SpellE>gfServerTrans</span>
- defines location of a translated server.</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>background -
defines the background image if any to display on web page</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span>company<span
style='mso-spacerun:yes'>    </span>- defines company name to display on web
page</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>   </span><span class=SpellE>tempDir</span><span
style='mso-spacerun:yes'>    </span>- where to put temporary files.<span
style='mso-spacerun:yes'>  </span>This path is relative to where the web</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>                </span>server
executes CGI scripts.<span style='mso-spacerun:yes'>  </span>It is good to
remove files that haven't</p>

<p class=MsoNormal><span style='mso-tab-count:2'>                                </span>been
accessed for 24 hours from this directory periodically, </p>

<p class=MsoNormal><span style='mso-tab-count:2'>                                </span>via
a <span class=SpellE>cron</span> job or similar mechanism.</p>

<p class=MsoNormal>The background and company commands are optional.<span
style='mso-spacerun:yes'>  </span>The <span class=SpellE>webBlat.cfg</span>
file must</p>

<p class=MsoNormal>have at least one valid <span class=SpellE>gfServer</span>
or <span class=SpellE>gfServerTrans</span> line, and a <span class=SpellE>tempDir</span>
line.</p>

<p class=MsoNormal>.<span style='mso-spacerun:yes'>  </span>Here is a <span
class=SpellE>webBlat.cfg</span> file that you</p>

<p class=MsoNormal>might find at a typical installation:</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>company Awesome Research Amalgamated</p>

<p class=MsoNormal>background /images/<span class=SpellE>dnaPaper.jpg</span></p>

<p class=MsoNormal><span class=SpellE>gfServer</span> <span class=SpellE>bigRamMachine</span>
17778 /data/genomes/2bit/hg16.2bit Human Genome</p>

<p class=MsoNormal><span class=SpellE>gfServer</span> <span class=SpellE>bigRamMachine</span>
17779 /data/genomes/2bit/hg16.2bit Human Genome</p>

<p class=MsoNormal><span class=SpellE>gfServer</span> <span class=SpellE>mouseServer</span>
17780 /data/genomes/2bit/mm4.2bit Mouse Genome</p>

<p class=MsoNormal><span class=SpellE>gfServer</span> <span class=SpellE>mouseServer</span>
17781 /data/genomes/2bit/mm4.2bit Mouse Genome</p>

<p class=MsoNormal><span class=SpellE>tempDir</span> ../trash</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>PUTTING WEBBLAT WHERE THE WEB SERVER CAN EXECUTE IT</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The details of this step vary highly from web server to web
server.<span style='mso-spacerun:yes'>  </span>On</p>

<p class=MsoNormal>a typical Apache installation it might be:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>ssh</span> <span class=SpellE>webServer</span></p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>cd</span> <span class=SpellE>kent/webBlat</span></p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>cp <span
class=SpellE>webBlat</span> <span class=SpellE>webBlat.cfg</span> /<span
class=SpellE>usr/local/apache/cgi</span>-bin</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>mkdir</span> /<span class=SpellE>usr</span>/local/apache/trash</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>chmod</span> 777 /<span class=SpellE>usr</span>/local/apache/trash</p>

<p class=MsoNormal>assuming that you've put the executable and <span
class=SpellE>config</span> file in <span class=SpellE>kent/webBlat</span>.</p>

<p class=MsoNormal>The program will create some files in the trash
directory.<span style='mso-spacerun:yes'>  </span>It is good to </p>

<p class=MsoNormal>periodically clean out old files from this directory.<span
style='mso-spacerun:yes'>   </span>On Mac OS-X instead</p>

<p class=MsoNormal>you might do:</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span>cp <span
class=SpellE>webBlat</span> <span class=SpellE>webBlat.cfg</span> /Library/<span
class=SpellE>WebServer</span>/CGI-Executables</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>mkdir</span> /Library/<span class=SpellE>WebServer</span>/trash</p>

<p class=MsoNormal><span style='mso-spacerun:yes'>     </span><span
class=SpellE>chmod</span> 777 /Library/<span class=SpellE>WebServer</span>/trash</p>

<p class=MsoNormal>Unless you are administering your own computer you will
likely need to ask your </p>

<p class=MsoNormal>local system administrators for help with this part of the <span
class=SpellE>webBlat</span> installation.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>File Formats<o:p></o:p></span></h3>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.<span class=SpellE>psl</span>
files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .<span class=SpellE>psl</span> file describes a series of
alignments in a dense easily parsed text format.<span
style='mso-spacerun:yes'>  </span>It begins with a five line header which
describes each field.<span style='mso-spacerun:yes'>  </span>Following this is
one line for each alignment with a tab between each field.<span
style='mso-spacerun:yes'>  </span>The fields are describe below in<span
style='mso-spacerun:yes'>  </span>a format suitable for many relational
databases.</p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>matches <span class=SpellE>int</span>
unsigned ,<span style='mso-spacerun:yes'>   </span><span style='mso-tab-count:
1'>        </span># Number of bases that match that aren't repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>misMatches</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-tab-count:1'>        </span>#
Number of bases that don't match<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>repMatches</span>
<span class=SpellE>int</span> unsigned , <span style='mso-tab-count:1'>       </span>#
Number of bases that match but are part of repeats<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>nCount</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>      
</span><span style='mso-tab-count:1'>     </span># Number of 'N' bases<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qNumInsert</span>
<span class=SpellE>int</span> unsigned , <span style='mso-tab-count:1'>       </span>#
Number of inserts in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qBaseInsert</span>
<span class=SpellE>int</span> unsigned , <span style='mso-tab-count:1'>      </span>#
Number of bases inserted in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tNumInsert</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'> 
</span><span style='mso-tab-count:1'>      </span># Number of inserts in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tBaseInsert</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'> 
</span><span style='mso-tab-count:1'>     </span># Number of bases inserted in
target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span>strand char(2) ,<span
style='mso-spacerun:yes'>            </span><span style='mso-tab-count:1'>     </span>#
+ or - for query strand, optionally followed by + or – for target strand<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qName</span>
<span class=SpellE>varchar</span>(255) ,<span style='mso-spacerun:yes'>      
</span><span style='mso-tab-count:1'>      </span># Query sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qSize</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>       
</span><span style='mso-tab-count:1'>     </span># Query sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qStart</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>    
</span><span style='mso-tab-count:1'>       </span># Alignment start position
in query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qEnd</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>        
</span><span style='mso-tab-count:1'>     </span># Alignment end position in
query<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tName</span>
<span class=SpellE>varchar</span>(255) ,<span style='mso-spacerun:yes'>      
</span><span style='mso-tab-count:1'>      </span># Target sequence name<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tSize</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>       
</span><span style='mso-tab-count:1'>     </span># Target sequence size<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>  </span><span
style='mso-spacerun:yes'>  </span><span class=SpellE>tStart</span> <span
class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>       </span><span
style='mso-tab-count:1'>     </span># Alignment start position in target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tEnd</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-spacerun:yes'>        
</span><span style='mso-tab-count:1'>     </span># Alignment end position in
target<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>blockCount</span>
<span class=SpellE>int</span> unsigned ,<span style='mso-tab-count:1'>        </span>#
Number of blocks in alignment. A block contains no gaps.<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>blockSizes</span>
<span class=SpellE>longblob</span> ,<span style='mso-spacerun:yes'>    </span><span
style='mso-tab-count:1'>        </span># Size of each block in a comma
separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>qStarts</span>
<span class=SpellE>longblob</span> ,<span style='mso-spacerun:yes'>  </span><span
style='mso-tab-count:1'>             </span># Start of each block in query in a
comma separated list<o:p></o:p></span></p>

<p class=MsoBodyTextIndent><span style='font-size:9.0pt;mso-bidi-font-size:
10.0pt'><span style='mso-spacerun:yes'>    </span><span class=SpellE>tStarts</span>
<span class=SpellE>longblob</span> ,<span style='mso-spacerun:yes'>  </span><span
style='mso-tab-count:1'>             </span># Start of each block in target in
a comma separated list<o:p></o:p></span></p>

<p class=MsoNormal>In general the coordinates in <span class=SpellE>psl</span>
files are “zero based half open.” The first base in a sequence is numbered zero
rather than one. When representing a range the end coordinate is not included
in the range. Thus the first 100 bases of a sequence are represented as 0-100,
and the second 100 bases are represented as 100-200. There is a another little
unusual feature in the .<span class=SpellE>psl</span> format. It has to do with
how coordinates are handled on the negative strand. In the <span class=SpellE>qStart/qEnd</span>
fields the coordinates are where it matches from the point of view of the
forward strand (even when the match is on the reverse strand). However on the <span
class=SpellE>qStarts</span>[] list, the coordinates are reversed. </p>

<p class=MsoNormal style='margin-right:-.25in'><o:p>&nbsp;</o:p></p>

<p class=MsoNormal style='margin-right:-.25in'>Here's an example of a 30-mer
that has 2 blocks that align on the minus strand and 2 blocks on the plus
strand (this sort of stuff happens in real life in response to assembly errors
sometimes). </p>

<p class=MsoNormal><span style='font-size:9.0pt;mso-bidi-font-size:10.0pt;
font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0<span
style='mso-spacerun:yes'>         </span>1<span
style='mso-spacerun:yes'>         </span>2<span
style='mso-spacerun:yes'>         </span>3 tens position in query<br>
0123456789012345678901234567890 ones position in query<br>
<span style='mso-spacerun:yes'>            </span>++++<span
style='mso-spacerun:yes'>          </span>+++++ plus strand alignment on query<br>
<span style='mso-spacerun:yes'>    </span>--------<span
style='mso-spacerun:yes'>    </span>----------<span
style='mso-spacerun:yes'>      </span>minus strand alignment on query <o:p></o:p></span></p>

<p class=MsoNormal>Plus strand:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>     </span><span
class=SpellE>qStart</span> 12 <span class=SpellE>qEnd</span> 31 <span
class=SpellE>blockSizes</span> 4,5 <span class=SpellE>qStarts</span> 12,26<br>
</span>Minus strand:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>     </span><span
class=SpellE>qStart</span> 4 <span class=SpellE>qEnd</span> 26 <span
class=SpellE>blockSizes</span> 10,8 <span class=SpellE>qStarts</span> 5,19 <o:p></o:p></span></p>

<p class=MsoNormal>Essentially the minus strand <span class=SpellE>blockSizes</span>
and <span class=SpellE>qStarts</span> are what you would get if you reverse
complemented the <span class=SpellE>query.However</span> the <span
class=SpellE>qStart</span> and <span class=SpellE>qEnd</span> are non-reversed.
To get from one to the other:<br>
<span style='font-family:Courier'><span style='mso-spacerun:yes'>     </span><span
class=SpellE>qStart</span> = <span class=SpellE>qSize</span> - <span
class=SpellE>revQEnd</span><br>
<span style='mso-spacerun:yes'>     </span><span class=SpellE>qEnd</span> = <span
class=SpellE>qSize</span> - <span class=SpellE>revQStart</span><br
style='mso-special-character:line-break'>
<![if !supportLineBreakNewLine]><br style='mso-special-character:line-break'>
<![endif]><o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.2bit files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .2bit file can store multiple DNA sequence (up to 4 gig
total) in a compact randomly accessible format.<span style='mso-spacerun:yes'> 
</span>The two<span style='mso-spacerun:yes'>  </span>bit files contain masking
information as well as the DNA itself.<span style='mso-spacerun:yes'>  
</span>The file begins with a 16 byte header containing the following fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>signature
– the number 0x1A412743 in the architecture of the machine that created the
file.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>version
– zero for now. Readers should abort if they see a version number higher than
0.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>sequenceCount</span> – the number of sequences in the file</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l1 level1 lfo1;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>4)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>reserved
– always zero for now.</p>

<p class=MsoNormal>All fields are 32 bits unless noted. If the signature value
is not as given, the reader program should byte swap the signature and see if
the swapped version matches.<span style='mso-spacerun:yes'>  </span>If so all
multiple-byte entities in the file will need to be byte-swapped.<span
style='mso-spacerun:yes'>  </span>This enables these binary files to be used
unchanged on different architectures.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The header is followed by a file index.<span
style='mso-spacerun:yes'>  </span>There is one entry in the index for each
sequence.<span style='mso-spacerun:yes'>   </span>Each index entry contains
three fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>nameSize</span> – a byte containing the length of the name field</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>name
– this contains the sequence name itself, and is variable length depending on <span
class=SpellE>nameSize</span>.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo2;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]>offset
– 32 bit offset of the sequence data relative to the start of the file</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal>The index is followed by the sequence records.<span
style='mso-spacerun:yes'>  </span>These contain 9 fields:</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>1)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>dnaSize</span> – number of bases of DNA in the sequence.</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>2)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>nBlockCount</span> – the number of blocks of N’s in the file
(representing unknown sequence).</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>3)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>nBlockStarts</span> – a starting position for each block of N’s</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>4)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>nBlockSizes</span> – the size of each block of N’s</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>5)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>maskBlockCount</span> – the number of masked (lower case) blocks</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>6)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>maskBlockStarts</span> – starting position for each masked block</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>7)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>maskBlockSizes</span> – the size of each masked block</p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l4 level1 lfo3;
tab-stops:list .5in'><![if !supportLists]><span style='mso-list:Ignore'>8)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span><![endif]><span
class=SpellE>packedDna</span> – the <span class=SpellE>dna</span> packed to two
bits per base as so: 00 – T,<span style='mso-spacerun:yes'>  </span>01 –
C,<span style='mso-spacerun:yes'>  </span>10 – A, 11 – G.<span
style='mso-spacerun:yes'>  </span>The first base is in the most significant 2
bits byte, and the last base in the least significant 2 bits, so that the
sequence TCAG would be represented as 00011011.<span style='mso-spacerun:yes'> 
</span>The <span class=SpellE>packedDna</span> field will be padded with 0 bits
as necessary<span style='mso-spacerun:yes'>  </span>so that it takes an even
multiple of 32 bit in the file, as this improves i/o performance on some
machines.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-size:12.0pt;mso-bidi-font-size:10.0pt'>.nib files<o:p></o:p></span></b></p>

<p class=MsoNormal>A .nib file describes a DNA sequence packing two bases into
each byte. Each nib file contains only a single sequence.<span
style='mso-spacerun:yes'>  </span>A nib file begins with a 32 bit signature
which is 0x6BE93D3A in the architecture of the machine that created the file,
and possibly a byte-swapped version of the same number on another machine.<span
style='mso-spacerun:yes'>  </span>This is followed by a 32 bit number in the
same format which describes the number of bases in the file.<span
style='mso-spacerun:yes'>   </span>This is followed by the bases themselves
packed two bases to the byte.<span style='mso-spacerun:yes'>  </span>The first
base is packed in the high order 4 bits (nibble), the second base in the low
order four bits.<span style='mso-spacerun:yes'>  </span>In C code:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>byte
= (base1&lt;&lt;4) + base2<o:p></o:p></span></p>

<p class=MsoNormal>The numerical values for the bases are:</p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-size:9.0pt;
mso-bidi-font-size:10.0pt;font-family:"Courier New";mso-bidi-font-family:"Times New Roman"'>0
– T,<span style='mso-spacerun:yes'>  </span>1 – C,<span
style='mso-spacerun:yes'>  </span>2 – A,<span style='mso-spacerun:yes'> 
</span>3 – G,<span style='mso-spacerun:yes'>  </span>4 – N (unknown)<o:p></o:p></span></p>

<p class=MsoNormal>The most significant bit in a nibble is set if the base is
masked. <span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'><o:p></o:p></span></p>

<h3><span style='font-size:16.0pt;mso-bidi-font-size:10.0pt'>Limits<o:p></o:p></span></h3>

<p class=MsoNormal>The <span class=SpellE>gfServer</span> program requires
approximately 1 byte for every 3 bases in the genome it is indexing in DNA
mode, and 1.5 bytes for each unmasked base in translated mode. The blat program
requires approximately two bytes for each base in the genome in DNA mode, and
three bytes for each base in translated mode. The other programs use relatively
little memory.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

</div>

</body>

</html>
