<H2>Description</H2>
<P>
This track shows $organism genome high-confidence gene annotations from the
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/" TARGET=_blank>Consensus 
Coding Sequence (CCDS) project</A>. This project is a collaborative effort 
to identify a core set of 
$organism protein-coding regions that are consistently annotated and of high 
quality. The long-term goal is to support convergence towards a standard set 
of gene annotations on the $organism genome.
</P>
<P>Collaborators include:
<UL>
<LI><A HREF="http://www.ebi.ac.uk/" TARGET=_blank>European Bioinformatics 
Institute</A> (EBI)
<LI><A HREF="http://www.ncbi.nlm.nih.gov" TARGET=_blank>National Center for 
Biotechnology Information</A> (NCBI)
<LI><A HREF="http://www.cbse.ucsc.edu/" TARGET=_blank>University of 
California, Santa Cruz</A> (UCSC)
<LI><A HREF="http://www.sanger.ac.uk/" TARGET=_blank>Wellcome Trust Sanger 
Institute</A> (WTSI)
</UL>

<H2>Methods</H2>
<P>
CDS annotations of the $organism genome were obtained from two sources:
<A HREF="http://www.ncbi.nlm.nih.gov/RefSeq/index.html" TARGET=_blank>NCBI 
RefSeq</A> and a union of the gene annotations from 
<A HREF="http://www.ensembl.org/" TARGET=_blank>Ensembl</A> and 
<A HREF="http://vega.sanger.ac.uk/" TARGET=_blank>Vega</A>, collectively known 
as <EM>Hinxton</EM>.</P>
<P>
Genes with identical CDS genomic coordinates in both sets become CCDS 
candidates. The genes undergo a quality evaluation, which must be approved by 
all collaborators. The following criteria are currently used to assess each
gene: 
<UL>
<LI> an initiating ATG (Exception: a non-ATG translation start codon is 
annotated if it has sufficient experimental support), a valid stop codon, and 
no in-frame stop codons (Exception: selenoproteins, which contain a TGA codon 
that is known to be translated to a selenocysteine instead of functioning as 
a stop codon) 
<LI> ability to be translated from the genome reference sequence without frameshifts
<LI> recognizable splicing sites
<LI> no intersection with putative pseudogene predictions
<LI> supporting transcripts and protein homology
<LI> conservation evidence with other species
</UL></P>
<P>
A unique CCDS ID is assigned to the CCDS, which links together all gene 
annotations with the same CDS.  CCDS gene annotations are under continuous
review, with periodic updates to this track.
</P>

<H2>Credits</H2>
<P>
This track was produced at UCSC from data downloaded from the
<A HREF="http://www.ncbi.nlm.nih.gov/CCDS/" TARGET=_blank>CCDS project</A> 
web site.
</P>

<H2>References</H2>
<P>
Pruitt KD, Harrow J, Harte RA, Wallin C, Diekhans M, Maglott DR, Searle S, 
Farrell CM, Loveland JE, Ruef BJ <EM>et al</EM>. 
<A HREF="http://genome.cshlp.org/content/early/2009/06/04/gr.080531.108.long"
TARGET=_blank>The consensus coding sequence (CCDS) project: Identifying a common protein-coding gene set for the human and mouse genomes</A>. 
<I>Genome Res.</I> 2009 Jun 4. [Epub ahead of print]
<P>
Hubbard T, Barker D, Birney E, Cameron G, Chen Y, Clark L, Cox T, Cuff J, Curwen V, Down T, <EM>et al</EM>.
<A HREF="http://nar.oupjournals.org/cgi/content/abstract/30/1/38" 
TARGET=_blank>The Ensembl genome database project</A>. 
<I>Nucl. Acids Res.</I> 2002 Jan 1;30(1):38-41.</P>
<P>
Pruitt KD, Tatusova T, Maglott DR.
<A HREF="http://nar.oupjournals.org/cgi/content/full/33/suppl_1/D501" 
TARGET=_blank>NCBI Reference Sequence (RefSeq): a curated non-redundant 
sequence database of genomes, transcripts and proteins</A>. 
<I>Nucl. Acids Res.</I> 2005 Jan 1;33(Database Issue):D501-D504. 
</P>
