<H2>Description</H2>
<P>
This track shows protein-coding gene predictions generated by 
<a href=http://contra.stanford.edu/contrast>CONTRAST</a>.  
Each predicted exon is colored according to confidence level: green (high
confidence), orange (medium confidence), or red (low confidence).
</P>

<H2>Methods</H2>
<P>
CONTRAST predicts protein-coding genes from a multiple genomic
alignment using a combination of discriminative machine learning techniques.  A two-stage approach is used, in which output from local classifiers is combined with a global model of gene structure.  CONTRAST is trained using a novel procedure designed to
maximize expected coding region boundary detection accuracy.

<P>
Please see the 
<a href=http://contra.stanford.edu/contrast>CONTRAST web site</a> for details on how these predictions were generated and an estimate of accuracy. 

<H2>Credits</H2>
<P>
Thanks to <a href=http://www.cs.stanford.edu/~ssgross>Samuel Gross</a> of the <a href=http://www.cs.stanford.edu/~batzoglou>Batzoglou lab</a> at Stanford University for providing these predictions.

<H2>References</H2>
<p>
Gross SS, Do CB, Sirota M, Batzoglou S.<A HREF="http://genomebiology.com/2007/8/12/R269"TARGET=_blank>
CONTRAST: A Discriminative, Phylogeny-free Approach to Multiple Informant De Novo Gene Prediction.</A>
<em>Genome Biology</em>. 2007 December;8(12):R269.
</p>
