<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2//EN">
<HTML>
<HEAD>
	<META HTTP-EQUIV="Content-Type" CONTENT="text/html;CHARSET=iso-8859-1">
	<META name="UCSC Genome Browser Links" content="UCSC Genome Browser"> 
	<META name="keywords" content="genome, genome browser, genome assembly, Blat, UCSC, bioinformatics, tools, gene prediction, SNP, EST, mRNA">
	<TITLE>UCSC Genome Browser: Best Links</TITLE>
	<LINK REL="STYLESHEET" HREF="style/HGStyle.css">
</HEAD>
<BODY BGCOLOR="FFF9D2" LINK="0000CC" VLINK="#330066" ALINK="#330066">
<A NAME="TOP"></A>


<TABLE BORDER=0 CELLPADDING=0 CELLSPACING=0 WIDTH="100%"><TR><TH COLSPAN=2 ALIGN="left"><IMG SRC="images/title.jpg" ALT="UCSC Genome Bioinformatics"></TH></TR>

<TR><TD COLSPAN=2 HEIGHT=40><!--#include virtual="inc/topbar.html"--></TD></TR>
<!--Content Tables------------------------------------------------------->
<TR><TD COLSPAN=2 CELLPADDING=10>	
<!--outer table is for border purposes-->
<A NAME="000"></A>
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8" BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
	<FONT SIZE="4"><B> &nbsp;UCSC Genome Bioinformatics Best Links </B></FONT> 
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH=100% CELLPADDING=0 BORDER=0><TH HEIGHT=10></TH><TR><TD WIDTH=3></TD><TD>
	This page contains links to useful databases,
	browsers, and tools.  The collection is primarily oriented
	toward human and mouse genomes. 
	To be included on this page, a tool must be 
	web-oriented, allow free access with no registration barriers, be usable 
	without installation or scripts, and be bioinformatic in nature. 
	<P>
	There are many internet sites 
	that provide good, comprehensive resource lists. 
	<A HREF="http://www.google.com" TARGET=_blank>Google</A> is also an
	excellent resource for searching for bioinformatics tools on the internet.
	Because of the alternative options available, this page is maintained 
	but is no 
	longer actively updated. The links were last 
	validated on 29 June 05. Additions and corrections should
	be sent to 
	<A HREF="mailto:&#103;e&#110;&#111;&#109;e&#45;&#97;&#100;m&#105;&#110;&#64;&#115;&#111;&#101;.&#117;&#99;sc.&#101;&#100;&#117;">&#103;e&#110;&#111;&#109;e&#45;&#97;&#100;m&#105;&#110;&#64;&#115;&#111;&#101;.&#117;&#99;sc.&#101;&#100;&#117;</A>.
	<!-- above address is genome-admin at soe.ucsc.edu -->

	</TD></TR><TR><TD COLSPAN=2>
	<TABLE CELLPADDING=0><TH HEIGHT=5></TH><TR><TD><UL><B>
  		<LI><A CLASS="toc" HREF="#111"> Main Databases </A></LI>
  		<LI><A CLASS="toc" HREF="#122"> Genome Browsers and Complete Genomes </A></LI>
  		<LI><A CLASS="toc" HREF="#133"> Gene Prediction and Feature Finding</A></LI>
  		<LI><A CLASS="toc" HREF="#144"> Protein Tools and Browsers </A></LI>
  		<LI><A CLASS="toc" HREF="#155"> Family Specific </A></LI>
  		<LI><A CLASS="toc" HREF="#166"> Blast, Blat, and Align Sequences </A></LI>
  		<LI><A CLASS="toc" HREF="#177"> Tool Collections </A></LI>
  		<LI><A CLASS="toc" HREF="#188"> Journals, Alerts, and General Internet </A></LI>
	</TD><TD VALIGN="top"></B></UL>
 	</TD></TR></TABLE>
	</TD></TR></TABLE>
	<BR></TD></TR></TABLE>
</TD></TR></TABLE>
	
	
<a name="111"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Main Databases</b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?CMD=Limits&DB=PubMed">PubMed</a> -- abstracts for 11 million biomedical articles
<li><a href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?db=Nucleotide">Entrez</a> -- gateway to nucleotide and protein sequences
<li><a href="http://www.ncbi.nlm.nih.gov/LocusLink/">LocusLink</A> -- curated sequence and gene descriptions
<li><a href="http://www.rcsb.org/">PDB</A> -- 3D protein database at RCSB
<li><a href="http://www.expasy.ch/cgi-bin/sprot-search-ful">SwissProt</A> -- keyword search annotated protein collection
<li><A HREF="http://www.expasy.org/srs5/">SRS</A> -- query Swiss sequence retrieval system
<li><A HREF="http://srs6.ebi.ac.uk/">EBI SRS</A> -- query session at EBI sequence retrieval center 
<li><a href="http://www.ebi.ac.uk/Databases/index.html">EMBL</A> -- gateway to EMBL sequence and structure databases
<li><a href="http://pbil.univ-lyon1.fr/databases/hovergen.html">Hovergen</A> -- homologous vertebrate genes database
<li><A HREF="http://zearth.kazusa.or.jp/huge/">HUGE</A> -- unidentified human proteins in KIAA series
<li><A HREF="http://srs-mips.gsf.de/">MIPS</A> -- gateway to protein information center in Munich
<li><A HREF="http://www.bioinf.man.ac.uk/dbbrowser/OWL/">OWL</A> -- non-redundant protein sequence database
<li><A HREF="http://www-nbrf.georgetown.edu/pir/searchdb.html">PIR</A> -- gateway to PIR protein database and query tools
<li><A HREF="http://www.ncbi.nlm.nih.gov/UniGene/query_tips.html">ChromList</A> -- lists UniGene clusters by chromosome
<li><A HREF="http://www.expasy.org/cgi-bin/lists?humchr12.txt">ChromProt</A> -- lists human proteins by chromosome
<li><A HREF="http://www.genome.ad.jp/kegg/kegg3.html">KEGG</A> -- Kyoto encyclopedia of pathways and maps
<li><A HREF="http://www.ncbi.nlm.nih.gov/Taxonomy/taxonomyhome.html">Taxonomy</A> -- taxonomic classes at NCBI
<li><A HREF="http://tolweb.org/tree/phylogeny.html">TreeLife</A> -- tree of life taxonomy project
<br><br>
<li><a href="http://www.gene.ucl.ac.uk/cgi-bin/nomenclature/searchgenes.pl">Hugo</A> -- recommended gene symbols at Genew3
<li><a href="http://www3.ncbi.nlm.nih.gov/omim/">OMIM 1</A>+
	<A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?CMD=Limits&DB=OMIM">2</A> -- simple and complex searches at OMIM
<li><a href="http://research.marshfieldclinic.org/genetics/">Marshfield</A> -- genetic and physical maps at Marshfield
<li><a href="http://archive.uwcm.ac.uk/uwcm/mg/hgmd0.html">HGMD Mut</A> -- human gene mutation database at Cardiff
<li><A HREF="http://www.ncbi.nlm.nih.gov:80/entrez/Omim/allresources.html#LocusSpecific">LocMutations</A> -- locus-specific mutation databases
<li><A HREF="http://www.mitomap.org">Mitomap</A> -- human mitochondrial genome database
<!--
<li><A HREF="http://www.ncbi.nlm.nih.gov/cgi-bin/Entrez/mzorder?tax=40674&name=Mammalia">Mito</A> -- mitochondrial genomes and alignment
-->
<li><A HREF="http://www.neuro.wustl.edu/neuromuscular/mitosyn.html">MitoWustl</A> -- mitochondrial genes, mutations and disorders
<li><A HREF="http://www.dhgp.de/research/projects/abstracts/9605.html">MitoP</A> -- mitochondrial genes, proteins and diseases from the Mitochondria Project
<li><A HREF="http://www-lecb.ncifcrf.gov/mitoDat/">MitoDat</A> -- nuclear genes specifying mitochondrial biogenesis
<br><br>
<li><a href="http://bioinfo.weizmann.ac.il/udb">UDB</A> -- integrated human genome map
<li><A HREF="http://www.ncbi.nlm.nih.gov/UniGene/query_tips.html">UniGene</A> -- query UniGene
<!--
<li><a href="http://www.ncbi.nlm.nih.gov/genome/seq/chr.cgi?CHR=20&SRT=ppos&MIN=0&ORG=Hs">Contigs</A> -- search NT contigs, dbSTS, GenMap99
-->
<li><a href="http://www-genome.wi.mit.edu/cgi-bin/contig/phys_map">Marker</A> -- search for marker, YACs, contigs
<li><A HREF="http://snp.cshl.org">SNP</A> -- single nucleotide polymorphisms at SNP Consortium
<li><A HREF="http://www.ncbi.nlm.nih.gov/SNP">dbSNP</A> -- search for single nucleotide polymorphisms at NCBI
<li><a href="http://headmaster.utmem.edu/search.html">WebQTL</a> -- QTL expression databases to explore gene interactions in recombinant inbred mice
<br><br>
<li><a href="http://uther.otago.ac.nz/Transterm.html">TransTerm</A> -- sequence context of stop and start codons
<li><a href="http://www.ch.embnet.org/software/ESTScan.html">ESTScan</A> -- detects coding regions despite frameshifts and low quality
<li><a href="http://www.cifn.unam.mx/Computational_Genomics/regulondb/">RegulonDB</A> -- transcription regulation database for various organisms
<li><A HREF="http://rebase.neb.com/rebase/rebase.html">Rebase</A> -- restriction enzyme cleavage sites
<li><A HREF="http://genomic.sanger.ac.uk/spldb/SpliceDB.html">SpliceDB</A> -- canonical and non-canonical splice sites in mammals
<li><a href="http://www.biology.ucsd.edu/~ipaulsen/transport/">MemTrans</A> -- genomic comparisons of membrane transport systems
<li><a href="http://www.Geneimprint.com">Geneimprint</A> -- imprinted gene database
<li><a href="http://www.kazusa.or.jp/codon/">Codon</A> -- codon usage database
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="122"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Genome Browsers & "Complete" Genomes </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="index.html">UCSC Genome Browser</A> -- human genome assembly and track browser at UCSC
<li><a href="http://www.ensembl.org/perl/mapview?chr=20">Ensembl</A> -- human genome and disease gene browser at EBI/Sanger	
<li><A HREF="http://www.ncbi.nlm.nih.gov/mapview/">Map Viewer</A> -- human genome assembly and map viewer at NCBI
<li><A HREF="../cgi-bin/hgText">UCSC Table
Browser</A> -- query human genome and annotation database tables
<li><A HREF="http://iubio.bio.indiana.edu:8089/man/">euGenes</A> -- five species map browser and 37,049 human gene reports
<li><A HREF="http://hgdownload.cse.ucsc.edu/downloads.html">Download</A> -- download human genome and install mirror
<li><A HREF="http://www-gsd.lbl.gov/vista">VISTA</A> -- portal for all VISTA comparative genomics tools 
<li><A HREF="http://pipeline.lbl.gov">VISTA Browser</A> -- whole genome alignment browser (human, mouse, rat, drosophila, etc.)
<li><A HREF="http://pipeline.lbl.gov/cgi-bin/GenomeVista">GenomeVISTA</A> --  align sequences to genome assemblies (human, mouse, etc.) 
<li><a href="http://ecrbrowser.dcode.org/">ECR Browser</A> -- visualization
and study of evolutionary relationships between vertebrate and non-vertebrate 
genomes
<li><A HREF="http://rvista.dcode.org/" TARGET=_blank>rVista</A> -- tool for 
finding potential regulatory elements in noncoding regions of the human 
genome 
<li><A HREF="http://www.tigr.org/tigr-scripts/CMR2/CMRHomePage.spl">TIGR CMR</A> -- comprehensive microbial resource and genome browser
<br><br>
<li><a href="http://www.ncbi.nlm.nih.gov/Genomes/index.html">NCBI</A> -- human and animal genome projects at NCBI
<li><a href="http://www.sanger.ac.uk/HGP/links.shtml">Sanger</A> -- human genome project at Sanger Centre
<li><a href="http://www.informatics.jax.org/">Jax</A> -- mouse genomics at Jackson Laboratory
<li><A HREF="http://ratmap.gen.gu.se/">RatMap</A> -- rat genome database at Goteborg			
<li><a href="http://zfish.uoregon.edu/">Zfin</A> -- zebrafish information network and database project
<li><a href="http://www.genoscope.cns.fr/externe/tetraodon/">Exofish</A> -- genome analysis of Tetraodon nigroviridis 
<li><a href="http://flybase.bio.indiana.edu/">Flybase</A> -- database of drosophila genomics at Indiana 
<li><A HREF="http://www.fruitfly.org">BDGP</A> -- fruit fly genomics center at Berkeley
<li><A HREF="http://www.wormbase.org/">WormBase</A> -- nematode mapping, sequencing and phenotypic repository
<li><a href="http://elegans.swmed.edu/genome.shtml">ACeDB</A> -- genetics and genomics resource launcher
<li><a href="http://genome-www.stanford.edu/Saccharomyces/">SGD</A> -- yeast genome database	
<li><A HREF="http://arabidopsis.org">TAIR</A> -- the arabidopsis information resource and tool center		
<li><a href="http://www.tigr.org/tdb/mdb/mdbcomplete.html">TIGR db</A> -- bacterial genomes at TIGR
<li><a href="http://www.ebi.ac.uk/genomes/mot/">GenTotals</A> -- monitoring tables for genome progress
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="133"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Gene Prediction and Feature Finding </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD>
<UL>
<li><a href="http://www.ncbi.nlm.nih.gov/IEB/Research/Acembly/index.html">Acembly/AceView</A> 
-- gene models reconstructed solely from mRNA and EST evidence 
<li><a href="http://genome.dkfz-heidelberg.de/cgi-bin/GENSCAN/genscan.cgi">Genscan D</A> +
	<a href="http://genes.mit.edu/GENSCAN.html">M</A> -- identification of complete gene structures in genomic DNA
<li><A HREF="http://genes.mit.edu/genomescan.html">GenomeScan</A> --  predicts genes incorporating protein homology
<li><A HREF="http://rgd.mcw.edu/">RGD</A> -- comparative mapping tool for rat-mouse-humany synteny
<li><a href="http://genomic.sanger.ac.uk/gf/gf.shtml">FGENESH</a> -- splice sites, coding exons, gene models, promoter and polyA
<li><a href="http://www.softberry.com/berry.phtml">Softberry</A> -- nucleotide sequence analysis, genes, promoters
<li><A HREF="http://genes.cs.wustl.edu/">TwinScan</A> -- gene prediction for eukaryotic genomic sequences
<li><A HREF="http://www.itba.mi.cnr.it/webgene/">WebGene</A> -- ten tools for analysis of protein-coding gene structure 
<li><A HREF="http://grail.lsd.ornl.gov/grailexp/">GrailEXP</A> -- exons, genes, promoters, polyA, CpG islands at ORNL 
<li><A HREF="http://www.sanger.ac.uk/Software/Wise2/genewiseform.shtml">GeneWise2</A> -- compares protein to genomic with introns and frameshifts 
<li><A HREF="http://www.fruitfly.org/seq_tools/genie.html">Genie</A> -- finds
multi-exon genes only, trained for Drosophila and other invertebrate organisms
<li><A HREF="http://genome.cs.mtu.edu/aat.html">AAT</A> -- analysis and annotation tool for finding genes 
<li><a href="http://www.cbs.dtu.dk/services/HMMgene/">HMMgene</a> -- prediction of vertebrate and nematode genes 
<li><a href="http://www.cbs.dtu.dk/services/NetGene2/">NetGene2</a> -- neural network predictions of splice sites 
<br><br>
<li><A HREF="http://www.ebi.ac.uk/genemark/">GeneMark</A> -- gene prediction server at EBI 
<li><a href="http://www1.imim.es/geneid.html">GeneID</a> -- gene prediction of selected signals and exons 
<li><a href="http://www.tbi.univie.ac.at/~ivo/RNA/">Vienna RNA</A> -- seven RNA analysis tools 
<br><br>
<li> <a href="http://searchlauncher.bcm.tmc.edu/seq-util/seq-util.html">Reverse Comp</A> -- reverse complement of DNA sequence 
<li> <A HREF="http://genes.mit.edu/McPromoter.html">McPromoter</A> -- finds eukaryotic RNA polymerase II transcription starts 
<li> <A HREF="http://bimas.dcrt.nih.gov/molbio/proscan/index.html">PromoterScan</A> -- predicts promoters using homologies with Pol II promoters	 	
<li> <A HREF="http://www.ncbi.nlm.nih.gov/gorf/gorf.html">ORF Finder</A> -- open reading frame finder for ORFs of cutoff size 
<li> <A HREF="http://www.ebi.ac.uk/emboss/cpgplot/">Emboss</A> -- predict CpG islands and isochores 
<li> <A HREF="http://bighost.area.ba.cnr.it/BIG/UTRHome/">UTR</A> -- untranslated regions of eukaryotic mRNAs

<li> <A HREF="http://www.ebi.ac.uk/asd/altsplice/" TARGET=_blank>AltSplice</A> -- database of alternate splicing in ESTs and disease genes 
<li> <a href="http://www.firstmarket.com/firstmarket/cutter/cut2.html">Webcutter</A> -- finds restriction endonuclease sites 
<! li> <! A HREF="http://ir2lcb.cnrs-mrs.fr/d_fsed/fsed.html"  FSED</A   -- frameshift error detection in new sequences  dead 12-02-03>
<li> <A HREF="http://www.bork.embl-heidelberg.de/ERR_WISE/">ERR_WISE</A> -- detection of frameshift sequencing errors  
<li> <A HREF="http://www.bork.embl-heidelberg.de/SEQERR/">SEQERR</A> -- detection of frameshift error in coding regions  
<li> <A HREF="http://www.ebi.ac.uk/cgi-bin/mutations/check.cgi">MutCheck</A> -- transcription and translation effect of mutations  
<li> <a href="http://blocks.fhcrc.org/~pauline/SIFT.html">Sift</A> -- predicts tolerated and untolerated protein substitutions  
<br><br>
<li> <a href="http://www.gene-regulation.com/">TransFac</A> -- transcription factor database and tool collection 
<li> <A HREF="http://www.gene-regulation.com">MatInspector</A> -- searches for transcription factor binding sites  
<li> <A HREF="http://www.bioinf.mdc-berlin.de/splice/">VarSplice</A> -- collection of annotated splice variants  
<li> <A HREF="http://bioweb.pasteur.fr/seqanal/interfaces/tfscan.html">TfScan</A> -- scans DNA sequences for transcription factors  
<br><br>
<li> <a href="http://ftp.genome.washington.edu/cgi-bin/RepeatMasker">RepeatMasker</A> -- finds retroposons and repeats  	
<li> <a href="http://www.girinst.org/Censor_Server-Data_Entry_Forms.html">Censor</A> -- finds repeated elements, Repbase updates, maps  
<li> <a href="http://www.biophys.uni-duesseldorf.de/local/DOTPLOT/dotplot.html">DotPlot</A> -- dot plot for internal DNA sequence repeats  
<li> <a href="http://www.bioinformaticssolutions.com/products/ph.php">PatternHunter</A> -- finds approximate repeat patterns in genomic DNA
<li> <a href="http://bibiserv.techfak.uni-bielefeld.de/cgi-bin/reputer_submit?mode=STARTUP">REPuter</A> 
	-- genomic repeats of all types to 100kbp
<li> <a href="http://www.biophys.uni-duesseldorf.de/local/TINOCO/tinoco.html">Tinoco</A> -- suggests secondary structure in RNA or DNA  
<li> <a href="http://www.csd.uwo.ca/~kzhang/rna/rna_align.html">RNA_align</A> 
	-- aligns two RNA species from secondary and tertiary structures
<li> <A HREF="http://www.embl-heidelberg.de/~andrade/papers/rep/search.html">REP</A> -- searches proteins for internal repeats  
<li> <A HREF="http://www.ebi.ac.uk/Radar/">Radar</A> -- detects and aligns protein repeats  
<li> <A HREF="http://ibivu.cs.vu.nl/programs/reprowww/">Repro</A> -- recognizes repeats in a single protein query 
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="144"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Protein Tools and Browsers </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="../cgi-bin/hgNear/">UCSC Gene Sorter</A> -- human and other proteins, can be grouped using many criteria
<li><a href="http://www.expasy.ch/tools/">SwissTools</A> -- large collection of protein tools and databases
<li><a href="http://www.expasy.ch/tools/dna.html">Translate</A> -- translates DNA into 6 frames 
<li><a href="http://www.ebi.ac.uk/cgi-bin/mutations/revtransl.cgi">ReverseTrans 1</A>+
	<a href="http://www.entelechon.com/eng/backtranslation.html">2</A> -- reverse-translates protein into DNA
<li><a href="http://www.ebi.ac.uk/saps/">SAPS</A> -- statistical analysis of protein sequences
<li><a href="http://www.ebi.ac.uk/emboss/pepinfo/">Emboss</A> -- peptide information and statistics
<br><br>
<li><a href="http://www.cbs.dtu.dk/services/SignalP/">Signalp</A> -- predicts signal peptide cleavage sites
<li><a href="http://psort.nibb.ac.jp/form.html">Psort</A> -- protein sorting signals
<li><a href="http://www.ch.embnet.org/software/TMPRED_form.html">TMpred</A> -- predicts transmembrane segments and orientation
<li><a href="http://www.cbs.dtu.dk/services/TMHMM/">TMHMM</A> -- predicts transmembrane helices
<li><a href="http://www.sbc.su.se/~erikw/toppred2/">TopPred2</A> -- topological prediction of membrane proteins
<li><a href="http://www.mbb.ki.se/tmap/index.html">	TMAP</A> -- predicts membrane proteins from multiple alignment		
<li><a
href="http://npsa-pbil.ibcp.fr/cgi-bin/npsa_automat.pl?page=npsa_predator.html">PREDATOR</A> -- secondary structure from multiple sequences
<li><a href="http://www.bork.embl-heidelberg.de/SSCP/">SSCP</A> -- predicts helix, strand and coil from composition
<li><a href="http://www.compbio.dundee.ac.uk/~www-jpred/">Jpred</A> -- consensus method for secondary structure prediction
<li><a href="http://www3.ebi.ac.uk/tops">TOPS</A> -- protein topology atlas and tool collection
<li> <A HREF="http://mathbio.nimr.mrc.ac.uk/tools/pops/">POPS</A> -- Parameter OPtimized surfaces
<br><br>
<li><a href="http://www.ebi.ac.uk/interpro/">Interpro</A> -- integrated resource of protein families, domains and sites
<li><a href="http://www.sanger.ac.uk/Pfam/">Pfam</A> -- alignments and hidden Markov models of protein domains
<li><a href="http://smart.embl-heidelberg.de/">SMART</A> -- simple modular architecture research tool
<li><a href="http://protein.toulouse.inra.fr/prodom.html">ProDom</A> -- protein domain database
<li><a href="http://us.expasy.org/tools/scanprosite/">ScanProsite</A> -- scans a sequence against PROSITE
<li><a href="http://hits.isb-sib.ch/cgi-bin/PFSCAN">ProfileScan</A> -- scans sequence against profile databases
<li><a href="http://www.ebi.ac.uk/printsscan/">FingerPRINTScan</A> -- search tool for Prints motifs
<li><a href="http://www.ebi.ac.uk/ppsearch/">ppsearch</A> -- pattern search tool for DNA or protein motifs
<li><a href="http://hmmer.wustl.edu/">HMMER</A>  --  profile hidden Markov models for sequence analysis
<li><a href="http://www.cse.ucsc.edu/research/compbio/sam.html">SAM</A>  --  sequence alignment and modeling system by HMM
<li><a href="http://blocks.fhcrc.org/">BLOCKS</A> -- aligns ungapped segments of conserved protein regions 
<li><a href="http://blocks.fhcrc.org/blocks/impala.html">IMPALA</A> -- blast search of BLOCKS conserved protein regions
<li><a href="http://bioweb.pasteur.fr/seqanal/motif/blimps-uk.html">BLIMPS</A> -- improved BLOCKS search tool
<li><a href="http://motif.genome.ad.jp/">Motif</A> -- searches for motifs in query sequence
<li><a href="http://fps.sdsc.edu/">FPS</A> -- family pairwise search library of protein families
<li><a href="http://scop.mrc-lmb.cam.ac.uk/scop/">SCOP</A> -- structural classification of proteins
<li><a href="http://supfam.mrc-lmb.cam.ac.uk/SUPERFAMILY/index.html">Superfamily</A> -- genome-wide structural classification of proteins
<li><a href="http://www.ebi.ac.uk/pratt/">Pratt</A> -- search for conserved patterns in protein sequences
<li><a href="http://www.biochem.ucl.ac.uk/bsm/cath_new/index.html">CATH</A> -- hierarchical classification of protein domain structures
<li><a href="http://meme.sdsc.edu/meme/website/mast.html">MAST</A> -- motif alignment search tool
<li><a href="http://meme.sdsc.edu/meme/website/meme.html">MEME</A> -- analyzes protein and DNA for motifs	
<li><a href="http://www.tigr.org/TIGRFAMs/">TIGRFAMs</A> -- protein families based on hidden Markov models
<br><br>
<li><a href="http://www.ebi.ac.uk/dali">DALI</A> -- coordinates of query compared to PDB database
<li><a href="http://genomic.sanger.ac.uk/123D/run123D.shtml">123D</A> -- predicts tertiary structure
<li><a href="http://www.ebi.ac.uk/dali/fssp">FSSP</A> -- fold classification based on structure
<li><a href="http://www.embl-heidelberg.de/predictprotein/predictprotein.html">PredProt</A> -- predicts protein structure
<li><a href="http://cubic.bioc.columbia.edu/predictprotein/">PredictProt</A> -- threading, secondary, solvent and transmembrane
<li><a href="http://dove.embl-heidelberg.de/3D/">Fold</A> -- 3D structures based on homologous sequences
<li><a href="http://www.ebi.ac.uk/dali/maxsprout/">MaxSprout</A> -- protein backbone and side chains from alpha coordinates
<li><a href="http://cl.sdsc.edu/ce/all-to-all/all-to-all-r.html">CE</A> -- finds structural alignments from PDB
<br><br>
<li><a href="http://pqs.ebi.ac.uk/">MSD PQS</A> -- protein quaternary structure query form at EBI
<li><a href="http://dip.doe-mbi.ucla.edu/dip/Search.cgi">DIP</A> -- database of interacting proteins
<br><br>
<li><a href="http://www-camis.stanford.edu/projects/helix/LPFC/">ProtFam</A> -- average core structures for families from structure
<li><a href="http://www.fccc.edu/research/labs/dunbrack/scwrl/">SCWRL</A> -- sidechain placement using a rotamer library	
<li><a href="http://www.bork.embl-heidelberg.de/ASC/">ASC</A> -- analytic surface calculation package
<!--
<li><a href="http://promoter.ics.uci.edu/BRNN-PRED/">AA Contacts</A> -- residue contacts, solvent accessibility
-->
<li><a href="http://prostar.carb.nist.gov/">ProStar</A> -- protein potential site
<br><br>
<li><a href="http://www.dkfz-heidelberg.de/spec/sweet2/doc/">Sweet</A> -- constructs 3D models of oligosaccharides from formula
<li><a href="http://www.glycosuite.com">GlycoSuiteDB</A> -- curated relational database of glycan structures	
<br><br>
<li><a href="http://www.expasy.ch/spdbv/">SwissPdbView</A> -- deep view for manipulation and analysis of PDB structures
<li><a href="http://www.rcsb.org/pdb/cgi/explore.cgi?job=graphics&pdbId=4HHB&page=&pid=11682995474077">PBD Viewers</A> -- choices of plug-ins and 3D structure viewers
<li><a href="http://www.proteinsociety.org/kinemages/kinpage.html">Kinemages</A> -- interactive 3D protein web display
<li><a href="http://www.umass.edu/microbio/rasmol/index2.htm">RasMol</A> -- molecular visualization resource
<li><a href="http://www.umass.edu/microbio/rasmol/">ProtExplorer</A> -- macromolecular viewer related to RasMol
<li><a href="http://www.umass.edu/microbio/chime">Chime</A> -- interactive web plug-in for 3D molecular structures
<li><a href="http://www.umass.edu/microbio/chime/morpher/">Morph</A> -- downloadable protein shape morpher
<li><a href="http://bioinformatics.org/combosa3d/index_b.html">COMBOSA3D</A> --  combines alignment, protein 3D, and Chime to color
<li><a href="http://www.ch.embnet.org/software/BOX_form.html">Boxshade</A> -- pretty shading of multiple alignments
<li><a href="http://www2.ebi.ac.uk/cgi-bin/translate/visprot.pl">ProtCol</A> -- tool for coloring proteins, default is by hydrophobicity
<li><a href="http://proteinexplorer.org/">ProtExpl</A> -- protein explorer interactive software
 </UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
	
<a name="155"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Family Specific </b></FONT></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.gpcr.org/7tm/">GPCRDB</A> -- G-protein coupled receptor database
<li><a href="http://www.receptors.org/NR/">NucleaRDB</A> -- nuclear receptors at UCSF
<li><a href="http://biochem1.basic-sci.georgetown.edu/nrr/NRR1.html">NRR1</A> -- nuclear receptor resource at Georgetown
<li><a href="http://bioinformatics.weizmann.ac.il/HORDE/">HORDE</A> -- human olfactory receptor genes
<li><a href="http://imgt.cines.fr">ImMunoGeneTics</A> -- vertebrate immunoglobulins (IG), T cell receptors (TR), and major histocompatibility complex (MHC)
<li><a href="http://www.cse.ucsc.edu/research/compbio/gpcr-subclass">GPCR</A> -- classifies G-protein coupled recepters
<li><a href="http://www-bio3d-igbmc.u-strasbg.fr/~wicker/Secator/secator.html">Secator</A> -- classifies any protein family (receptors, tRNA synthetases)
<li><a href="http://merops.sanger.ac.uk/" TARGET=_blank>Merops</A> -- curated database on proteases
<li><a href="http://www.mad-cow.org/00/sci_archive_frame.html">Prion</A> -- annotated prion-doppel gene family
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="166"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Blast, Blat, and Align Sequences </FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov/blast/index.nojs.cgi">Blast</A>  --  java-free NCBI blast launch
<li><a href="http://www.expasy.org/cgi-bin/BLASTEMBnet-CH.pl">SIB Blast</A>  --  blastp and tBlastn HTGS, UniGene at EMBnet
<!--
<li><a href="http://www.ncbi.nlm.nih.gov/gorf/bl2.html">Blast2</A>  --  restricts blast to user query and target
-->
<li><a href="http://www.proweb.org/proweb/Tools/WU-blast.html">WU-Blast</A>  --  blast query against user fasta database
<li><a href="http://searchlauncher.bcm.tmc.edu/downloads/software/batch_client.html">Batch</A>  --  batch blast from Baylor
<br><br>
<li><a href="../cgi-bin/hgBlat?command=start">Blat</A>  --  quick alignment of query to human genome assembly
<li><a href="http://www.ncbi.nlm.nih.gov/genome/seq/page.cgi?F=HsBlast.html&&ORG=Hs">HumFin</A>  --  blast finished human genome 
chromosome
<li><a href="http://mouse.ensembl.org/">Mus 1</A>+
	<a href="http://www.ncbi.nlm.nih.gov/genome/seq/page.cgi?F=MmBlast.html&&ORG=Mm">2</A> 
	--  blast finished mouse genome by chromosome
<li><a href="http://www.ensembl.org/perl/blastview">Ensembl</A>  --  blast human genome project
<li><a href="http://www.sanger.ac.uk/DataSearch">Sanger</A>  --  blast sequencing center chromosomes
<li><a href="http://www.ncbi.nlm.nih.gov/blast/mmtrace.html">Traces</A>  --  blast against raw new genomic trace reads
<li><a href="http://trace.ensembl.org/perl/ssahaview">SSAHA</A>  --  trace server at Ensembl
<li><a href="http://www.hri.co.jp/HUNT">Hunt</A>  --  search human novel transcript cDNAs
<br><br>
<li><a href="http://www.cse.ucsc.edu/~kent/xenoAli/xenAliTwo.html">WABA</A>  --  genomic to genomic comparisons
<li><a href="http://bio.cse.psu.edu/pipmaker/">Pipmaker</A>  --  percent identity plot alignment of two genomic sequences
<li><a href="http://www.sanger.ac.uk/Software/Wise2/dbaform.shtml">DBA</A>  --  aligns two sequences sharing conserved blocks
<li><a href="http://www.sanger.ac.uk/Software/Alfresco/">Alfresco</A>  --  comparative genome sequence analysis
<br><br>
<li><a href="http://www2.ebi.ac.uk/clustalw/">ClustalW</A>  --  aligns multiple sequences			
<li><a href="http://www.toulouse.inra.fr/multalin.html">MultAlin</A>  --  aligns multiple sequences with flexible formatting
<li><a href="http://www.ch.embnet.org/software/LALIGN_form.html">Lalign</A>  --  find multiple matching subsegments in two sequences
<li><a href="http://www.bork.embl-heidelberg.de/Alignment/consensus.html">Consensus</A>  --  calculates consensus to ClustalW alginment
<li><a href="http://www.ebi.ac.uk/fasta33/">Fasta3</A>  --  fasta3 alignment of protein sequence
<li><a href="http://www.ebi.ac.uk/snpfasta3/">SNP-Fasta3</A>  --  alignment to HgBase SNP consortium
<li><a href="http://www.ebi.ac.uk/MPsrch/">MPsrch</A>  --  alignment tool with many parameter controls
<li><a href="http://www.ebi.ac.uk/scanps/">ScanPS</A>  --  multi-alignment of query protein sequence
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="177"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Tool Collections </FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.genome.gov/">NHGRI</A>,
	<a href="http://www.ensembl.org/genome/central/">EBI</A>,
	<a href="http://www.ncbi.nlm.nih.gov/Genomes/index.html">NCBI</A>
	-- "Genome Hub" pages
<li><a href="http://www.ncbi.nlm.nih.gov/Sitemap/index.html">NCBI</A>  --  site map to extensive services
<li><a href="http://searchlauncher.bcm.tmc.edu/">BCM</A>  --  launches numerous nucleotide and protein tools
<li><a href="http://bioinfolab.unl.edu/emlab/links.html">CMS</A>  --  extensive directory of tools by category
<li><a href="http://www.up.univ-mrs.fr/~wabim/english/logligne.html">ABI</A>  --  very large French tool compilation
<li><a href="http://www3.oup.co.uk/nar/database/c/">NAR</A> -- curated tool collection at Nucleic Acid Research
<li><a href="http://glamdring.ucsd.edu/others/dsmith/dnasys.html#contents">UCSD</A> -- genome analysis by task area
<li><a href="http://workbench.sdsc.edu/">Workbench</A> -- unified area for searching and analysis
<li><a href="http://www.ebi.ac.uk/Tools/index.html">EMBL1</A>  --  collection of database searching and tools
<li><a href="http://www.embl-heidelberg.de/Services/index.html">EMBL2</A>  --  various EMBL alignment tools
<li><a href="http://picasso.ucsf.edu/mol_str_links.html">UCSF1</A>+<a href="http://www.cmpharm.ucsf.edu/cohen/links/">2</A> collections of tool links at UCSF
<li><a href="http://www.bioinf.mdc-berlin.de/">MDC</A>  --  bioinformatics tools at Delbruck Center
<li><a href="http://www.ebi.ac.uk/Tools/index.html">EBI</A>  --  tool and utility collection at EBI
<li><a href="http://www.sdsc.edu/Software/molbio.html">SDSC</A>  --  tools at San Diego Supercomputer Center
<li><a href="http://www.bork.embl-heidelberg.de/">Bork</A>  --  ten tool server of Bork lab
<li><a href="http://www.ebi.ac.uk/biocat/biocat.html">Biocatalog</A>  --  bioinformatic software collection
<li><a href="http://mia.sdsc.edu">MIA</A>  --  search agent finds macromolecular information
<li><a href="http://evolution.genetics.washington.edu/phylip/software.html">Phylip</A>  --  tools and software for phylogenies
<li><a href="http://www.hgmp.mrc.ac.uk/Registered/Menu/phylo.html">Phylo</A>  --  provides 14 utilties for phylogenetic analysis
<li><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">PAML</A>  --  -- maximum likelihood of evolutionary parameters and
phylogenies
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>
	
<a name="188"><BR></a>
<!--outer table is for border purposes-->
<TABLE WIDTH="100%" BGCOLOR="#888888" BORDER="0" CELLSPACING="0" CELLPADDING="1"><TR><TD>	
    <TABLE BGCOLOR="fffee8" WIDTH="100%"  BORDER="0" CELLSPACING="0" CELLPADDING="0"><TR><TD>	
	<TABLE BGCOLOR="D9E4F8"  BACKGROUND="images/hr.gif" WIDTH=100%><TR><TD>
		<!--cheap trick to get background on row to be continuous-->
		<TABLE CELLPADDING=0 CELLSPACING=0 MARGIN=0 WIDTH=100%><TR><TD ALIGN="left">
			<FONT SIZE="4" ><b>&nbsp; Journals, Alerts, and General Internet</FONT></b></TD>
			<TD ALIGN="right"><A HREF="#000"><IMG SRC="images/index.gif" ALT="Index->" BORDER=0></A>
		</TD></TR></TABLE>
	</TD></TR></TABLE>
	<TABLE BGCOLOR="fffee8" WIDTH="100%" CELLPADDING=0><TH HEIGHT=10></TH><TR><TD><UL>
<li><a href="http://www.ncbi.nlm.nih.gov:80/entrez/jrbrowser.cgi">PubMedJB</A>  --  journal browser allows access to current issues
<li><a href="http://pubcrawler.gen.tcd.ie/">PubCrawler</A>  --  provides alerts to new abstracts at PubMed
<li><a href="http://highwire.stanford.edu/">HighWire</A>  --  publishes online journals and provides alerts
<br><br>				
<li><a href="http://www.hum-molgen.de/journals/">New Issues</A>  --  provides entry to 35 genomics jounals
<li><a href="http://genomebiology.com/home/">GenomeBio</A>  --  free access to post-genomics journal, pre-print server	
<li><a href="http://www.pnas.org/">PNAS</A>  --  free access to month-old articles in PNAS
<li><a href="http://www.jbc.org">JBC</A>  --  nearly free access to most cited journal
<li><a href="http://www.sciencemag.org/">Science</A>  --  high circulation general science journal
<li><a href="http://www.nature.com/">Nature</A>  --  gateway to restricted spin-off journals
<li><a href="http://www.ajhg.org">AmJHumGen</A>  --  leading disease gene annotation journal
<li><a href="http://www.academicpress.com/genomics">Genomics</A>  --  genomics articles sometimes stale
<li><a href="http://www.genome.org/">GenomeRes</A>  --  genomic research and methods articles
<li><a href="http://www.public.iastate.edu/~pedro/rt_journals.html">Pedro</A>  --  large journal (and tool) collection
<br><br>
<li><a href="http://cancerweb.ncl.ac.uk/omd/">OMD</A>  --  online medical dictionary
<li><a href="http://www.google.com/">Google </A>  --  best internet search site for web tools
<li><a href="http://www.lycos.com/">Lycos</A>  --  widely used general search tool
<li><a href="http://dir.yahoo.com/Health/News_and_Media/">Yahoo</A>  --  internet categories and news
<li><a href="http://www.altavista.com/r?F09">LangTrans</A>  --  useful for rough translations between languages
</UL></TD></TR></TABLE>
	</TD></TR></TABLE>
</TD></TR></TABLE>	

</TD></TR></TABLE>
</BODY></HTML>
