<H2>Description</H2>
<P>
The miRNA track shows microRNAs from the
<A HREF="http://microrna.sanger.ac.uk/" TARGET=_blank>
miRBase</A> at <A HREF="http://www.sanger.ac.uk/" TARGET=_blank>The 
Wellcome Trust Sanger Institute</A>. </P>

<H2>Display Conventions and Configuration</H2>
<P>
The precursor forms of
microRNAs (mirs) in the sense
orientation are shown in black; those in the reverse orientation are colored
grey.</P>
<P>
To display only those items that exceed a specific unnormalized score, enter
a minimum score between 0 and 1000 in the text box at the top of the track 
description page. For this track, a score of 960 signifies that the miRNA is on the + strand and a score of 480 signifies that it is on the - strand. 
</P>

<H2>Methods</H2>
<P>
Precursor miRNA genomic locations from
<a href="http://microrna.sanger.ac.uk/" target="_blank">
miRBase</a>
were calculated using wublastn for sequence alignment with the requirement of 
100% identity. The extents of the precursor sequences were not generally known,
and were predicted based on base-paired hairpin structure. miRBase is 
described in Griffiths-Jones, S. <EM>et al</EM>. (2006). The miRNA Registry 
is described in Griffiths-Jones, S. (2004) and Weber, M.J. (2005) in the 
References section below.</p>

<H2>Credits</H2>
<P>
Genome coordinates for this track were obtained from the miRBase sequences 
<A HREF="ftp://ftp.sanger.ac.uk/pub/mirbase/sequences/CURRENT/genomes/" 
TARGET=_BLANK>FTP site</A>.
</P>
<H2>References</H2>
<P>
When making use of these data, please cite the folowing articles in addition to 
the primary sources of the miRNA sequences:</P>
<P>
Griffiths-Jones S, Saini HK, van Dongen S, Enright AJ.
<A HREF="http://nar.oxfordjournals.org/cgi/content/full/36/suppl_1/D154" 
TARGET=_BLANK>miRBase: tools for microRNA genomics</A>. 
<EM>Nucleic Acids Res.</EM> 2008 Jan 1;36(Database issue):D154-8.</P>
<P>
Griffiths-Jones S, Grocock RJ, van Dongen S, Bateman A, Enright AJ.
<A HREF="http://nar.oxfordjournals.org/cgi/content/abstract/34/suppl_1/D140" 
TARGET=_BLANK>miRBase: microRNA sequences, targets and gene nomenclature</A>.
<EM>Nucleic Acids Res.</EM> 2006 Jan 1;34(Database issue):D140-4.</P>
<P>
Griffiths-Jones S. 
<A HREF="http://nar.oupjournals.org/cgi/content/abstract/32/suppl_1/D109"
TARGET=_blank>The microRNA Registry</A>.
<EM>Nucleic Acids Res.</EM> 2004 Jan 1;32(Database issue):D109-11.</P>
<P>
Weber MJ. 
<A HREF="http://www.blackwell-synergy.com/doi/abs/10.1111/j.1432-1033.2004.04389.x"
TARGET=_blank>New human and mouse microRNA genes found by homology search</A>.
<em>Febs J.</em> 2005 Jan;272(1):59-73.</P>
<P>
You may also want to cite The Wellcome Trust Sanger Institute 
<A HREF="http://www.sanger.ac.uk/Software/Rfam/mirna/"
TARGET=_blank>miRNA Registry</A>.</P>
<P>
The following publication provides guidelines on miRNA annotation:
<BR>Ambros V, Bartel B, Bartel DP, Burge CB, Carrington JC, Chen X,
Dreyfuss G, Eddy SR, Griffiths-Jones S, Marshall M <EM>et al</EM>.
<A HREF="http://www.rnajournal.org/cgi/content/abstract/9/3/277" 
TARGET=_blank>A uniform system for microRNA annotation</A>.  
<EM>RNA.</EM> 2003 Mar;9(3):277-9.</P>
<P>
