<H2>Methods</H2>
<P>
The plot represents relative mononucleosome enrichment at each 
position in the genome (on a log of 2 scale). The coverage metric is 
given according to a formula [ (1+n)/N ] / [ (1+c)/C ] where n and c 
are the numbers of putative 147 base pair cores covering each base pair from 
nucleosome and control data, N and C are the total number of 
nucleosome and control reads obtained by SOLiD sequencing with 25 base pairs 
mapped to the reference genome.
</P>

<H2>Data download</H2>
<P>
The source data for this track can be fetched from our 
<A HREF="ftp://hgdownload.cse.ucsc.edu/goldenPath/$db/nucleosome/"
TARGET=_blank>ftp server</A>
(<EM>nucleosomeAdjustedCoverage.wigAscii.gz</EM>; 450 Mb compressed, 
1.6 Gb uncompressed).<BR>
This data is formatted into the two-column <EM>variableStep</EM>
<A HREF="http://genome.ucsc.edu/goldenPath/help/wiggle.html"
TARGET=_blank>wiggle format</A>.
</P>

<H2>Credits</H2>
The data for this track is supplied by the
<A HREF="http://mendel.stanford.edu/SidowLab/" TARGET=_blank>Sidow Lab,</A>
and the
<A HREF="http://firelab.stanford.edu/" TARGET=_blank>Fire Lab</A>
at the <A HREF="http://med.stanford.edu/"
TARGET=_blank>Stanford School of Medicine.</A><BR>
Track display advice provided by
<A HREF="mailto:&#104;&#105;&#114;a&#109;&#64;&#115;&#111;&#101;
.&#117;&#99;&#115;&#99;.&#101;&#100;u">Hiram Clawson</A>,
UCSC Genome Browser Engineering.


<H2>References</H2>
<P>
Valouev A, Ichikawa J, Tonthat T, Stuart J, Ranade S, Peckham H, Zeng K,
Malek JA, Costa G, McKernan K, Sidow A, Fire A, Johnson SM.
<A HREF="http://genome.cshlp.org/cgi/content/short/18/7/1051"
TARGET=_blank>A high-resolution, nucleosome position map of
C. elegans reveals a lack of universal
sequence-dictated positioning</A>.
<EM>Genome Res</EM>. 2008 Jul;18(7):1051-63. Epub 2008 May 13.
</P>
<P>
Supported oligo ligation detection (SOLiD) sequencing technology from
<A HREF="http://solid.appliedbiosystems.com/"
TARGET=_blank>Applied Biosystems.</A>
</P>
